// @aws-sdk/client-rekognition/dist-cjs/schemas/schemas_0.js
const _A = "Asset";
const _AB = "AggregateBy";
const _AC = "AutoCreate";
const _ADE = "AccessDeniedException";
const _AF = "AssociatedFace";
const _AFL = "AssociatedFacesList";
const _AFR = "AssociateFacesRequest";
const _AFRs = "AssociateFacesResponse";
const _AFs = "AssociatedFaces";
const _AFss = "AssociateFaces";
const _AI = "AuditImage";
const _AIL = "AuditImagesLimit";
const _AIu = "AuditImages";
const _AM = "AudioMetadata";
const _AML = "AudioMetadataList";
const _AR = "AgeRange";
const _AU = "AutoUpdate";
const _Al = "Aliases";
const _Ar = "Arn";
const _As = "Assets";
const _At = "Attributes";
const _B = "Bytes";
const _BB = "BoundingBox";
const _BBC = "BoundingBoxCount";
const _BF = "BlackFrame";
const _BMV = "BaseModelVersion";
const _BP = "BodyParts";
const _BTTIS = "BillableTrainingTimeInSeconds";
const _Ba = "Background";
const _Be = "Beard";
const _Bl = "Blue";
const _Br = "Brightness";
const _Bu = "Bucket";
const _C = "Code";
const _CA = "CollectionArn";
const _CARN = "CollectionARN";
const _CBP = "CoversBodyPart";
const _CC = "ContentClassifiers";
const _CCR = "CreateCollectionRequest";
const _CCRr = "CreateCollectionResponse";
const _CCr = "CreateCollection";
const _CD = "CelebrityDetail";
const _CDR = "CreateDatasetRequest";
const _CDRr = "CreateDatasetResponse";
const _CDr = "CreateDataset";
const _CE = "ConflictException";
const _CF = "ComparedFace";
const _CFC = "CustomizationFeatureConfig";
const _CFCMC = "CustomizationFeatureContentModerationConfig";
const _CFL = "ComparedFaceList";
const _CFLS = "CreateFaceLivenessSession";
const _CFLSR = "CreateFaceLivenessSessionRequest";
const _CFLSRS = "CreateFaceLivenessSessionRequestSettings";
const _CFLSRr = "CreateFaceLivenessSessionResponse";
const _CFM = "CompareFacesMatch";
const _CFML = "CompareFacesMatchList";
const _CFR = "CompareFacesRequest";
const _CFRo = "CompareFacesResponse";
const _CFUL = "CompareFacesUnmatchList";
const _CFe = "CelebrityFaces";
const _CFo = "CompareFaces";
const _CH = "ConnectedHome";
const _CHFU = "ConnectedHomeForUpdate";
const _CHS = "ConnectedHomeSettings";
const _CHSFU = "ConnectedHomeSettingsForUpdate";
const _CI = "CollectionId";
const _CIo = "CollectionIds";
const _CL = "CustomLabel";
const _CLe = "CelebrityList";
const _CLo = "ContainsLabels";
const _CLu = "CustomLabels";
const _CM = "ContentModeration";
const _CMD = "ContentModerationDetection";
const _CMDo = "ContentModerationDetections";
const _CP = "ChallengePreference";
const _CPR = "CreateProjectRequest";
const _CPRr = "CreateProjectResponse";
const _CPV = "CopyProjectVersion";
const _CPVR = "CopyProjectVersionRequest";
const _CPVRo = "CopyProjectVersionResponse";
const _CPVRr = "CreateProjectVersionRequest";
const _CPVRre = "CreateProjectVersionResponse";
const _CPVr = "CreateProjectVersion";
const _CPh = "ChallengePreferences";
const _CPr = "CreateProject";
const _CR = "CelebrityRecognition";
const _CRT = "ClientRequestToken";
const _CRe = "CelebrityRecognitions";
const _CRo = "ColorRange";
const _CSIF = "ComparedSourceImageFace";
const _CSP = "CreateStreamProcessor";
const _CSPR = "CreateStreamProcessorRequest";
const _CSPRr = "CreateStreamProcessorResponse";
const _CSSC = "CSSColor";
const _CT = "ContentTypes";
const _CTo = "ContentType";
const _CTom = "CompletionTimestamp";
const _CTon = "ConfidenceThreshold";
const _CTr = "CreationTimestamp";
const _CU = "CreateUser";
const _CUR = "CreateUserRequest";
const _CURr = "CreateUserResponse";
const _Ca = "Categories";
const _Ce = "Celebrity";
const _Cel = "Celebrities";
const _Ch = "Challenge";
const _Cha = "Changes";
const _Co = "Codec";
const _Con = "Confidence";
const _Cont = "Contrast";
const _D = "Datasets";
const _DA = "DatasetArn";
const _DAa = "DataAttributes";
const _DAe = "DetectionAttributes";
const _DC = "DatasetChanges";
const _DCL = "DetectCustomLabels";
const _DCLR = "DetectCustomLabelsRequest";
const _DCLRe = "DetectCustomLabelsResponse";
const _DCR = "DeleteCollectionRequest";
const _DCRe = "DeleteCollectionResponse";
const _DCRes = "DescribeCollectionRequest";
const _DCResc = "DescribeCollectionResponse";
const _DCe = "DeleteCollection";
const _DCes = "DescribeCollection";
const _DCo = "DominantColors";
const _DCom = "DominantColor";
const _DD = "DatasetDescription";
const _DDE = "DistributeDatasetEntries";
const _DDER = "DistributeDatasetEntriesRequest";
const _DDERi = "DistributeDatasetEntriesResponse";
const _DDML = "DistributeDatasetMetadataList";
const _DDR = "DeleteDatasetRequest";
const _DDRe = "DeleteDatasetResponse";
const _DDRes = "DescribeDatasetRequest";
const _DDResc = "DescribeDatasetResponse";
const _DDe = "DeleteDataset";
const _DDes = "DescribeDataset";
const _DDi = "DistributeDataset";
const _DE = "DatasetEntries";
const _DF = "DeletedFaces";
const _DFL = "DisassociatedFacesList";
const _DFR = "DeleteFacesRequest";
const _DFRe = "DeleteFacesResponse";
const _DFRet = "DetectFacesRequest";
const _DFRete = "DetectFacesResponse";
const _DFRi = "DisassociateFacesRequest";
const _DFRis = "DisassociateFacesResponse";
const _DFe = "DetectionFilter";
const _DFel = "DeleteFaces";
const _DFet = "DetectFaces";
const _DFi = "DisassociatedFace";
const _DFis = "DisassociatedFaces";
const _DFisa = "DisassociateFaces";
const _DFu = "DurationFrames";
const _DL = "DetectLabels";
const _DLD = "DatasetLabelDescription";
const _DLDa = "DatasetLabelDescriptions";
const _DLIB = "DetectLabelsImageBackground";
const _DLIF = "DetectLabelsImageForeground";
const _DLIP = "DetectLabelsImageProperties";
const _DLIPS = "DetectLabelsImagePropertiesSettings";
const _DLIQ = "DetectLabelsImageQuality";
const _DLR = "DetectLabelsRequest";
const _DLRe = "DetectLabelsResponse";
const _DLS = "DatasetLabelStats";
const _DLSe = "DetectLabelsSettings";
const _DM = "DurationMillis";
const _DML = "DetectModerationLabels";
const _DMLR = "DetectModerationLabelsRequest";
const _DMLRe = "DetectModerationLabelsResponse";
const _DMLa = "DatasetMetadataList";
const _DMa = "DatasetMetadata";
const _DP = "DeleteProject";
const _DPA = "DestinationProjectArn";
const _DPE = "DetectProtectiveEquipment";
const _DPER = "DetectProtectiveEquipmentRequest";
const _DPERe = "DetectProtectiveEquipmentResponse";
const _DPP = "DeleteProjectPolicy";
const _DPPR = "DeleteProjectPolicyRequest";
const _DPPRe = "DeleteProjectPolicyResponse";
const _DPR = "DeleteProjectRequest";
const _DPRe = "DeleteProjectResponse";
const _DPRes = "DescribeProjectsRequest";
const _DPResc = "DescribeProjectsResponse";
const _DPV = "DeleteProjectVersion";
const _DPVR = "DeleteProjectVersionRequest";
const _DPVRe = "DeleteProjectVersionResponse";
const _DPVRes = "DescribeProjectVersionsRequest";
const _DPVResc = "DescribeProjectVersionsResponse";
const _DPVe = "DescribeProjectVersions";
const _DPe = "DescribeProjects";
const _DS = "DatasetSource";
const _DSMPTE = "DurationSMPTE";
const _DSP = "DataSharingPreference";
const _DSPFU = "DataSharingPreferenceForUpdate";
const _DSPR = "DeleteStreamProcessorRequest";
const _DSPRe = "DeleteStreamProcessorResponse";
const _DSPRes = "DescribeStreamProcessorRequest";
const _DSPResc = "DescribeStreamProcessorResponse";
const _DSPe = "DeleteStreamProcessor";
const _DSPes = "DescribeStreamProcessor";
const _DSa = "DatasetStats";
const _DT = "DatasetType";
const _DTF = "DetectTextFilters";
const _DTR = "DetectTextRequest";
const _DTRe = "DetectTextResponse";
const _DTe = "DetectedText";
const _DTet = "DetectText";
const _DU = "DeleteUser";
const _DUR = "DeleteUserRequest";
const _DURe = "DeleteUserResponse";
const _E = "Emotions";
const _EC = "EntryCount";
const _ED = "EquipmentDetection";
const _EDq = "EquipmentDetections";
const _EDy = "EyeDirection";
const _EE = "ErrorEntries";
const _EFN = "EndFrameNumber";
const _EII = "ExternalImageId";
const _EO = "EyeOpen";
const _EOy = "EyesOpen";
const _ER = "EvaluationResult";
const _ETM = "EndTimestampMillis";
const _ETSMPTE = "EndTimecodeSMPTE";
const _Em = "Emotion";
const _Ey = "Eyeglasses";
const _F = "Face";
const _FA = "FaceAttributes";
const _FC = "FeatureConfig";
const _FCa = "FaceCount";
const _FD = "FaceDetails";
const _FDA = "FlowDefinitionArn";
const _FDL = "FaceDetailList";
const _FDa = "FaceDetail";
const _FDac = "FaceDetection";
const _FDace = "FaceDetections";
const _FDai = "FailureDetails";
const _FH = "FrameHeight";
const _FI = "FaceId";
const _FIa = "FaceIds";
const _FL = "FaceList";
const _FM = "FaceMatches";
const _FML = "FaceMatchList";
const _FMT = "FaceMatchThreshold";
const _FMV = "FaceModelVersion";
const _FMVa = "FaceModelVersions";
const _FMa = "FaceMatch";
const _FN = "FragmentNumber";
const _FO = "FaceOccluded";
const _FR = "FaceRecord";
const _FRL = "FaceRecordList";
const _FRa = "FaceRecords";
const _FRr = "FrameRate";
const _FS = "F1Score";
const _FSS = "FaceSearchSettings";
const _FSa = "FaceSearch";
const _FW = "FrameWidth";
const _Fa = "Faces";
const _Fe = "Feature";
const _Fea = "Features";
const _Fi = "Filters";
const _Fo = "Foreground";
const _For = "Format";
const _G = "Geometry";
const _GCI = "GetCelebrityInfo";
const _GCIR = "GetCelebrityInfoRequest";
const _GCIRe = "GetCelebrityInfoResponse";
const _GCM = "GetContentModeration";
const _GCMR = "GetContentModerationRequest";
const _GCMRM = "GetContentModerationRequestMetadata";
const _GCMRe = "GetContentModerationResponse";
const _GCR = "GetCelebrityRecognition";
const _GCRR = "GetCelebrityRecognitionRequest";
const _GCRRe = "GetCelebrityRecognitionResponse";
const _GFD = "GetFaceDetection";
const _GFDR = "GetFaceDetectionRequest";
const _GFDRe = "GetFaceDetectionResponse";
const _GFLSR = "GetFaceLivenessSessionResults";
const _GFLSRR = "GetFaceLivenessSessionResultsRequest";
const _GFLSRRe = "GetFaceLivenessSessionResultsResponse";
const _GFS = "GetFaceSearch";
const _GFSR = "GetFaceSearchRequest";
const _GFSRe = "GetFaceSearchResponse";
const _GL = "GeneralLabels";
const _GLD = "GetLabelDetection";
const _GLDR = "GetLabelDetectionRequest";
const _GLDRM = "GetLabelDetectionRequestMetadata";
const _GLDRe = "GetLabelDetectionResponse";
const _GLS = "GeneralLabelsSettings";
const _GMAJ = "GetMediaAnalysisJob";
const _GMAJR = "GetMediaAnalysisJobRequest";
const _GMAJRe = "GetMediaAnalysisJobResponse";
const _GPT = "GetPersonTracking";
const _GPTR = "GetPersonTrackingRequest";
const _GPTRe = "GetPersonTrackingResponse";
const _GRM = "GetRequestMetadata";
const _GSD = "GetSegmentDetection";
const _GSDR = "GetSegmentDetectionRequest";
const _GSDRe = "GetSegmentDetectionResponse";
const _GT = "GroundTruth";
const _GTD = "GetTextDetection";
const _GTDR = "GetTextDetectionRequest";
const _GTDRe = "GetTextDetectionResponse";
const _GTM = "GroundTruthManifest";
const _Ge = "Gender";
const _Gr = "Green";
const _H = "High";
const _HC = "HexCode";
const _HE = "HasErrors";
const _HLA = "HumanLoopArn";
const _HLACER = "HumanLoopActivationConditionsEvaluationResults";
const _HLAO = "HumanLoopActivationOutput";
const _HLAR = "HumanLoopActivationReasons";
const _HLC = "HumanLoopConfig";
const _HLDA = "HumanLoopDataAttributes";
const _HLN = "HumanLoopName";
const _HLQEE = "HumanLoopQuotaExceededException";
const _He = "Height";
const _I = "Id";
const _IF = "IndexFaces";
const _IFMV = "IndexFacesModelVersion";
const _IFR = "IndexFacesRequest";
const _IFRn = "IndexFacesResponse";
const _II = "ImageId";
const _IIFE = "InvalidImageFormatException";
const _IME = "InvalidManifestException";
const _IP = "ImageProperties";
const _IPE = "InvalidParameterException";
const _IPME = "IdempotentParameterMismatchException";
const _IPRIE = "InvalidPolicyRevisionIdException";
const _IPTE = "InvalidPaginationTokenException";
const _IQ = "ImageQuality";
const _ISE = "InternalServerError";
const _ISOE = "InvalidS3ObjectException";
const _ITLE = "ImageTooLargeException";
const _Im = "Image";
const _In = "Input";
const _Ind = "Index";
const _Ins = "Instance";
const _Inst = "Instances";
const _JI = "JobId";
const _JN = "JobName";
const _JS = "JobStatus";
const _JT = "JobTag";
const _KDS = "KinesisDataStream";
const _KG = "KnownGender";
const _KKI = "KmsKeyId";
const _KP = "KeyPrefix";
const _KVS = "KinesisVideoStream";
const _KVSSS = "KinesisVideoStreamStartSelector";
const _KVSSSS = "KVSStreamStartSelector";
const _L = "Logref";
const _LA = "LabelAlias";
const _LAa = "LabelAliases";
const _LC = "LabelCategory";
const _LCEF = "LabelCategoryExclusionFilters";
const _LCIF = "LabelCategoryInclusionFilters";
const _LCR = "ListCollectionsRequest";
const _LCRi = "ListCollectionsResponse";
const _LCa = "LabelCategories";
const _LCi = "ListCollections";
const _LD = "LabelDetection";
const _LDE = "ListDatasetEntries";
const _LDER = "ListDatasetEntriesRequest";
const _LDERi = "ListDatasetEntriesResponse";
const _LDL = "ListDatasetLabels";
const _LDLR = "ListDatasetLabelsRequest";
const _LDLRi = "ListDatasetLabelsResponse";
const _LDS = "LabelDetectionSettings";
const _LDa = "LabelDetections";
const _LE = "LabeledEntries";
const _LEE = "LimitExceededException";
const _LEF = "LabelExclusionFilters";
const _LF = "ListFaces";
const _LFR = "ListFacesRequest";
const _LFRi = "ListFacesResponse";
const _LIB = "LivenessImageBlob";
const _LIF = "LabelInclusionFilters";
const _LMAJ = "ListMediaAnalysisJobs";
const _LMAJR = "ListMediaAnalysisJobsRequest";
const _LMAJRi = "ListMediaAnalysisJobsResponse";
const _LMV = "LabelModelVersion";
const _LN = "LabelName";
const _LOC = "LivenessOutputConfig";
const _LPP = "ListProjectPolicies";
const _LPPR = "ListProjectPoliciesRequest";
const _LPPRi = "ListProjectPoliciesResponse";
const _LS = "LabelStats";
const _LSP = "ListStreamProcessors";
const _LSPR = "ListStreamProcessorsRequest";
const _LSPRi = "ListStreamProcessorsResponse";
const _LTFR = "ListTagsForResource";
const _LTFRR = "ListTagsForResourceRequest";
const _LTFRRi = "ListTagsForResourceResponse";
const _LU = "ListUsers";
const _LUR = "ListUsersRequest";
const _LURi = "ListUsersResponse";
const _LUT = "LastUpdatedTimestamp";
const _LUTa = "LastUpdateTimestamp";
const _La = "Landmarks";
const _Lab = "Labels";
const _Labe = "Label";
const _Label = "Labeled";
const _Lan = "Landmark";
const _Le = "Left";
const _Lo = "Low";
const _M = "Message";
const _MADMLC = "MediaAnalysisDetectModerationLabelsConfig";
const _MAI = "MediaAnalysisInput";
const _MAJ = "MediaAnalysisJobs";
const _MAJD = "MediaAnalysisJobDescription";
const _MAJDe = "MediaAnalysisJobDescriptions";
const _MAJFD = "MediaAnalysisJobFailureDetails";
const _MAMS = "MediaAnalysisManifestSummary";
const _MAMV = "MediaAnalysisModelVersions";
const _MAOC = "MediaAnalysisOperationsConfig";
const _MAOCe = "MediaAnalysisOutputConfig";
const _MAR = "MediaAnalysisResults";
const _MBBH = "MinBoundingBoxHeight";
const _MBBW = "MinBoundingBoxWidth";
const _MC = "MatchConfidence";
const _MCP = "MinCoveragePercentage";
const _MCi = "MinConfidence";
const _MDC = "MaxDominantColors";
const _MDIS = "MaxDurationInSeconds";
const _MF = "MaxFaces";
const _MIU = "MinInferenceUnits";
const _MIUa = "MaxInferenceUnits";
const _ML = "ModerationLabel";
const _MLa = "MaxLabels";
const _MLo = "ModerationLabels";
const _MMV = "ModerationModelVersion";
const _MO = "MouthOpen";
const _MPDE = "MalformedPolicyDocumentException";
const _MPT = "MaxPixelThreshold";
const _MR = "MaxResults";
const _MS = "ManifestSummary";
const _MSC = "MinSegmentConfidence";
const _MU = "MatchedUser";
const _MUa = "MaxUsers";
const _MV = "ModelVersions";
const _MVo = "ModelVersion";
const _Ma = "Maximum";
const _Mi = "Minimum";
const _Mo = "Moderation";
const _Mu = "Mustache";
const _N = "Name";
const _NC = "NotificationChannel";
const _NOC = "NumberOfChannels";
const _NT = "NextToken";
const _O = "Output";
const _OC = "OutputConfig";
const _OCp = "OperationsConfig";
const _OCr = "OrientationCorrection";
const _OI = "OptIn";
const _P = "Pose";
const _PA = "ProjectArn";
const _PD = "ProjectDescriptions";
const _PDe = "PersonDetail";
const _PDer = "PersonDetection";
const _PDers = "PersonDetections";
const _PDo = "PolicyDocument";
const _PDr = "ProjectDescription";
const _PEBP = "ProtectiveEquipmentBodyPart";
const _PEMV = "ProtectiveEquipmentModelVersion";
const _PEP = "ProtectiveEquipmentPerson";
const _PEPr = "ProtectiveEquipmentPersons";
const _PES = "ProtectiveEquipmentSummary";
const _PESA = "ProtectiveEquipmentSummarizationAttributes";
const _PI = "PersonsIndeterminate";
const _PIa = "ParentId";
const _PM = "PersonMatch";
const _PMe = "PersonMatches";
const _PN = "ProjectName";
const _PNa = "ParentName";
const _PNo = "PolicyName";
const _PNr = "ProjectNames";
const _PP = "PixelPercent";
const _PPP = "PutProjectPolicy";
const _PPPR = "PutProjectPolicyRequest";
const _PPPRu = "PutProjectPolicyResponse";
const _PPr = "ProjectPolicies";
const _PPro = "ProjectPolicy";
const _PRI = "PolicyRevisionId";
const _PT = "ProducerTimestamp";
const _PTD = "ParametersToDelete";
const _PTEE = "ProvisionedThroughputExceededException";
const _PV = "ProjectVersion";
const _PVA = "ProjectVersionArn";
const _PVD = "ProjectVersionDescriptions";
const _PVDr = "ProjectVersionDescription";
const _PWRE = "PersonsWithRequiredEquipment";
const _PWREe = "PersonsWithoutRequiredEquipment";
const _Pa = "Parents";
const _Par = "Parent";
const _Pe = "Persons";
const _Per = "Person";
const _Pi = "Pitch";
const _Po = "Polygon";
const _Poi = "Point";
const _Q = "Quality";
const _QC = "QuotaCode";
const _QF = "QualityFilter";
const _R = "Red";
const _RA = "RoleArn";
const _RAEE = "ResourceAlreadyExistsException";
const _RAe = "ResourceArn";
const _RC = "RecognizeCelebrities";
const _RCR = "RecognizeCelebritiesRequest";
const _RCRe = "RecognizeCelebritiesResponse";
const _RET = "RequiredEquipmentTypes";
const _RI = "ReferenceImage";
const _RIUE = "ResourceInUseException";
const _RNFE = "ResourceNotFoundException";
const _RNRE = "ResourceNotReadyException";
const _ROI = "RegionsOfInterest";
const _ROIFU = "RegionsOfInterestForUpdate";
const _ROIe = "RegionOfInterest";
const _RT = "ResourceType";
const _Re = "Results";
const _Rea = "Reasons";
const _Ro = "Roll";
const _S = "Smile";
const _SA = "SummarizationAttributes";
const _SB = "SortBy";
const _SBu = "S3Bucket";
const _SC = "ServiceCode";
const _SCM = "StartContentModeration";
const _SCMR = "StartContentModerationRequest";
const _SCMRt = "StartContentModerationResponse";
const _SCR = "StartCelebrityRecognition";
const _SCRR = "StartCelebrityRecognitionRequest";
const _SCRRt = "StartCelebrityRecognitionResponse";
const _SCi = "SimplifiedColor";
const _SCt = "StatusCode";
const _SD = "S3Destination";
const _SDe = "SegmentDetection";
const _SDeg = "SegmentDetections";
const _SF = "SearchedFace";
const _SFBB = "SearchedFaceBoundingBox";
const _SFBI = "SearchFacesByImage";
const _SFBIR = "SearchFacesByImageRequest";
const _SFBIRe = "SearchFacesByImageResponse";
const _SFC = "SearchedFaceConfidence";
const _SFD = "SearchedFaceDetails";
const _SFDR = "StartFaceDetectionRequest";
const _SFDRt = "StartFaceDetectionResponse";
const _SFDt = "StartFaceDetection";
const _SFI = "SearchedFaceId";
const _SFN = "StartFrameNumber";
const _SFR = "SearchFacesRequest";
const _SFRe = "SearchFacesResponse";
const _SFS = "StartFaceSearch";
const _SFSR = "StartFaceSearchRequest";
const _SFSRt = "StartFaceSearchResponse";
const _SFU = "SettingsForUpdate";
const _SFe = "SearchFaces";
const _SFh = "ShotFilter";
const _SI = "SourceImage";
const _SIF = "SourceImageFace";
const _SIOC = "SourceImageOrientationCorrection";
const _SIe = "SessionId";
const _SJHLACER = "SynthesizedJsonHumanLoopActivationConditionsEvaluationResults";
const _SKP = "S3KeyPrefix";
const _SLD = "StartLabelDetection";
const _SLDR = "StartLabelDetectionRequest";
const _SLDRt = "StartLabelDetectionResponse";
const _SM = "StatusMessage";
const _SMAJ = "StartMediaAnalysisJob";
const _SMAJR = "StartMediaAnalysisJobRequest";
const _SMAJRt = "StartMediaAnalysisJobResponse";
const _SMC = "StatusMessageCode";
const _SNFE = "SessionNotFoundException";
const _SNSTA = "SNSTopicArn";
const _SO = "S3Object";
const _SP = "StreamProcessors";
const _SPA = "SourceProjectArn";
const _SPAt = "StreamProcessorArn";
const _SPDSP = "StreamProcessorDataSharingPreference";
const _SPI = "StreamProcessorInput";
const _SPL = "StreamProcessorList";
const _SPNC = "StreamProcessorNotificationChannel";
const _SPO = "StreamProcessorOutput";
const _SPS = "StreamProcessorSettings";
const _SPSFU = "StreamProcessorSettingsForUpdate";
const _SPSS = "StreamProcessingStartSelector";
const _SPSSt = "StreamProcessingStopSelector";
const _SPT = "StartPersonTracking";
const _SPTR = "StartPersonTrackingRequest";
const _SPTRt = "StartPersonTrackingResponse";
const _SPV = "StartProjectVersion";
const _SPVA = "SourceProjectVersionArn";
const _SPVR = "StartProjectVersionRequest";
const _SPVRt = "StartProjectVersionResponse";
const _SPVRto = "StopProjectVersionRequest";
const _SPVRtop = "StopProjectVersionResponse";
const _SPVt = "StopProjectVersion";
const _SPt = "StreamProcessor";
const _SQEE = "ServiceQuotaExceededException";
const _SR = "SampleRate";
const _SRC = "SourceRefContains";
const _SS = "ShotSegment";
const _SSD = "StartSegmentDetection";
const _SSDF = "StartSegmentDetectionFilters";
const _SSDFt = "StartShotDetectionFilter";
const _SSDR = "StartSegmentDetectionRequest";
const _SSDRt = "StartSegmentDetectionResponse";
const _SSP = "StartStreamProcessor";
const _SSPR = "StartStreamProcessorRequest";
const _SSPRt = "StartStreamProcessorResponse";
const _SSPRto = "StopStreamProcessorRequest";
const _SSPRtop = "StopStreamProcessorResponse";
const _SSPt = "StopStreamProcessor";
const _SST = "SelectedSegmentTypes";
const _SSt = "StartSelector";
const _SSto = "StopSelector";
const _ST = "SimilarityThreshold";
const _STCDF = "StartTechnicalCueDetectionFilter";
const _STD = "StartTextDetection";
const _STDF = "StartTextDetectionFilters";
const _STDR = "StartTextDetectionRequest";
const _STDRt = "StartTextDetectionResponse";
const _STI = "SegmentTypeInfo";
const _STIe = "SegmentTypesInfo";
const _STM = "StartTimestampMillis";
const _STSMPTE = "StartTimecodeSMPTE";
const _STe = "SegmentTypes";
const _SU = "SearchedUser";
const _SUBI = "SearchUsersByImage";
const _SUBIR = "SearchUsersByImageRequest";
const _SUBIRe = "SearchUsersByImageResponse";
const _SUR = "SearchUsersRequest";
const _SURe = "SearchUsersResponse";
const _SUe = "SearchUsers";
const _Se = "Settings";
const _Seg = "Segments";
const _Sh = "Sharpness";
const _Si = "Similarity";
const _St = "Status";
const _Su = "Summary";
const _Sun = "Sunglasses";
const _T = "Top";
const _TCF = "TechnicalCueFilter";
const _TCS = "TechnicalCueSegment";
const _TD = "TrainingData";
const _TDL = "TextDetectionList";
const _TDR = "TrainingDataResult";
const _TDRe = "TestingDataResult";
const _TDRex = "TextDetectionResult";
const _TDRext = "TextDetectionResults";
const _TDe = "TestingData";
const _TDex = "TextDetections";
const _TDext = "TextDetection";
const _TE = "ThrottlingException";
const _TET = "TrainingEndTimestamp";
const _TEo = "TotalEntries";
const _TI = "TargetImage";
const _TIOC = "TargetImageOrientationCorrection";
const _TK = "TagKeys";
const _TL = "TotalLabels";
const _TLa = "TaxonomyLevel";
const _TMV = "TextModelVersion";
const _TR = "TagResource";
const _TRR = "TagResourceRequest";
const _TRRa = "TagResourceResponse";
const _Ta = "Tags";
const _Ti = "Timestamp";
const _Ty = "Type";
const _U = "Urls";
const _UC = "UserCount";
const _UDE = "UpdateDatasetEntries";
const _UDER = "UpdateDatasetEntriesRequest";
const _UDERp = "UpdateDatasetEntriesResponse";
const _UF = "UnmatchedFaces";
const _UFA = "UnsuccessfulFaceAssociations";
const _UFAL = "UnsuccessfulFaceAssociationList";
const _UFAn = "UnsuccessfulFaceAssociation";
const _UFD = "UnsuccessfulFaceDeletions";
const _UFDL = "UnsuccessfulFaceDeletionsList";
const _UFDLn = "UnsuccessfulFaceDisassociationList";
const _UFDn = "UnsuccessfulFaceDisassociations";
const _UFDns = "UnsuccessfulFaceDeletion";
const _UFDnsu = "UnsuccessfulFaceDisassociation";
const _UFL = "UnsearchedFacesList";
const _UFn = "UnindexedFaces";
const _UFni = "UnindexedFace";
const _UFnr = "UnrecognizedFaces";
const _UFns = "UnsearchedFaces";
const _UFnse = "UnsearchedFace";
const _UI = "UserId";
const _UL = "UserList";
const _UM = "UserMatches";
const _UML = "UserMatchList";
const _UMT = "UserMatchThreshold";
const _UMs = "UserMatch";
const _UR = "UntagResource";
const _URR = "UntagResourceRequest";
const _URRn = "UntagResourceResponse";
const _US = "UserStatus";
const _USP = "UpdateStreamProcessor";
const _USPR = "UpdateStreamProcessorRequest";
const _USPRp = "UpdateStreamProcessorResponse";
const _Us = "Users";
const _Use = "User";
const _V = "Value";
const _VD = "VersionDescription";
const _VDa = "ValidationData";
const _VM = "VideoMetadata";
const _VML = "VideoMetadataList";
const _VN = "VersionName";
const _VNe = "VersionNames";
const _VTLE = "VideoTooLargeException";
const _Va = "Validation";
const _Ve = "Version";
const _Ver = "Versions";
const _Vi = "Video";
const _W = "Width";
const _WF = "WordFilter";
const _X = "X";
const _Y = "Yaw";
const _Y_ = "Y";
const _a = "application/json";
const _c = "client";
const _e = "error";
const _hE = "httpError";
const _mT = "mediaType";
const _s = "smithy.ts.sdk.synthetic.com.amazonaws.rekognition";
const _se = "server";
const n0 = "com.amazonaws.rekognition";
const { TypeRegistry } = require("@smithy/core/schema");
const { AccessDeniedException, ConflictException, HumanLoopQuotaExceededException, IdempotentParameterMismatchException, ImageTooLargeException, InternalServerError, InvalidImageFormatException, InvalidManifestException, InvalidPaginationTokenException, InvalidParameterException, InvalidPolicyRevisionIdException, InvalidS3ObjectException, LimitExceededException, MalformedPolicyDocumentException, ProvisionedThroughputExceededException, ResourceAlreadyExistsException, ResourceInUseException, ResourceNotFoundException, ResourceNotReadyException, ServiceQuotaExceededException, SessionNotFoundException, ThrottlingException, VideoTooLargeException } = require("../models/errors");
const { RekognitionServiceException } = require("../models/RekognitionServiceException");
const _s_registry = TypeRegistry.for(_s);
const RekognitionServiceException$ = [-3, _s, "RekognitionServiceException", 0, [], []];
exports.RekognitionServiceException$ = RekognitionServiceException$;
_s_registry.registerError(RekognitionServiceException$, RekognitionServiceException);
const n0_registry = TypeRegistry.for(n0);
const AccessDeniedException$ = [-3, n0, _ADE,
    { [_e]: _c },
    [_M, _C, _L],
    [0, 0, 0]
];
exports.AccessDeniedException$ = AccessDeniedException$;
n0_registry.registerError(AccessDeniedException$, AccessDeniedException);
const ConflictException$ = [-3, n0, _CE,
    { [_e]: _c },
    [_M, _C, _L],
    [0, 0, 0]
];
exports.ConflictException$ = ConflictException$;
n0_registry.registerError(ConflictException$, ConflictException);
const HumanLoopQuotaExceededException$ = [-3, n0, _HLQEE,
    { [_e]: _c, [_hE]: 402 },
    [_RT, _QC, _SC, _M, _C, _L],
    [0, 0, 0, 0, 0, 0]
];
exports.HumanLoopQuotaExceededException$ = HumanLoopQuotaExceededException$;
n0_registry.registerError(HumanLoopQuotaExceededException$, HumanLoopQuotaExceededException);
const IdempotentParameterMismatchException$ = [-3, n0, _IPME,
    { [_e]: _c },
    [_M, _C, _L],
    [0, 0, 0]
];
exports.IdempotentParameterMismatchException$ = IdempotentParameterMismatchException$;
n0_registry.registerError(IdempotentParameterMismatchException$, IdempotentParameterMismatchException);
const ImageTooLargeException$ = [-3, n0, _ITLE,
    { [_e]: _c },
    [_M, _C, _L],
    [0, 0, 0]
];
exports.ImageTooLargeException$ = ImageTooLargeException$;
n0_registry.registerError(ImageTooLargeException$, ImageTooLargeException);
const InternalServerError$ = [-3, n0, _ISE,
    { [_e]: _se },
    [_M, _C, _L],
    [0, 0, 0]
];
exports.InternalServerError$ = InternalServerError$;
n0_registry.registerError(InternalServerError$, InternalServerError);
const InvalidImageFormatException$ = [-3, n0, _IIFE,
    { [_e]: _c },
    [_M, _C, _L],
    [0, 0, 0]
];
exports.InvalidImageFormatException$ = InvalidImageFormatException$;
n0_registry.registerError(InvalidImageFormatException$, InvalidImageFormatException);
const InvalidManifestException$ = [-3, n0, _IME,
    { [_e]: _c },
    [_M, _C, _L],
    [0, 0, 0]
];
exports.InvalidManifestException$ = InvalidManifestException$;
n0_registry.registerError(InvalidManifestException$, InvalidManifestException);
const InvalidPaginationTokenException$ = [-3, n0, _IPTE,
    { [_e]: _c },
    [_M, _C, _L],
    [0, 0, 0]
];
exports.InvalidPaginationTokenException$ = InvalidPaginationTokenException$;
n0_registry.registerError(InvalidPaginationTokenException$, InvalidPaginationTokenException);
const InvalidParameterException$ = [-3, n0, _IPE,
    { [_e]: _c },
    [_M, _C, _L],
    [0, 0, 0]
];
exports.InvalidParameterException$ = InvalidParameterException$;
n0_registry.registerError(InvalidParameterException$, InvalidParameterException);
const InvalidPolicyRevisionIdException$ = [-3, n0, _IPRIE,
    { [_e]: _c },
    [_M, _C, _L],
    [0, 0, 0]
];
exports.InvalidPolicyRevisionIdException$ = InvalidPolicyRevisionIdException$;
n0_registry.registerError(InvalidPolicyRevisionIdException$, InvalidPolicyRevisionIdException);
const InvalidS3ObjectException$ = [-3, n0, _ISOE,
    { [_e]: _c },
    [_M, _C, _L],
    [0, 0, 0]
];
exports.InvalidS3ObjectException$ = InvalidS3ObjectException$;
n0_registry.registerError(InvalidS3ObjectException$, InvalidS3ObjectException);
const LimitExceededException$ = [-3, n0, _LEE,
    { [_e]: _c },
    [_M, _C, _L],
    [0, 0, 0]
];
exports.LimitExceededException$ = LimitExceededException$;
n0_registry.registerError(LimitExceededException$, LimitExceededException);
const MalformedPolicyDocumentException$ = [-3, n0, _MPDE,
    { [_e]: _c },
    [_M, _C, _L],
    [0, 0, 0]
];
exports.MalformedPolicyDocumentException$ = MalformedPolicyDocumentException$;
n0_registry.registerError(MalformedPolicyDocumentException$, MalformedPolicyDocumentException);
const ProvisionedThroughputExceededException$ = [-3, n0, _PTEE,
    { [_e]: _c },
    [_M, _C, _L],
    [0, 0, 0]
];
exports.ProvisionedThroughputExceededException$ = ProvisionedThroughputExceededException$;
n0_registry.registerError(ProvisionedThroughputExceededException$, ProvisionedThroughputExceededException);
const ResourceAlreadyExistsException$ = [-3, n0, _RAEE,
    { [_e]: _c },
    [_M, _C, _L],
    [0, 0, 0]
];
exports.ResourceAlreadyExistsException$ = ResourceAlreadyExistsException$;
n0_registry.registerError(ResourceAlreadyExistsException$, ResourceAlreadyExistsException);
const ResourceInUseException$ = [-3, n0, _RIUE,
    { [_e]: _c },
    [_M, _C, _L],
    [0, 0, 0]
];
exports.ResourceInUseException$ = ResourceInUseException$;
n0_registry.registerError(ResourceInUseException$, ResourceInUseException);
const ResourceNotFoundException$ = [-3, n0, _RNFE,
    { [_e]: _c },
    [_M, _C, _L],
    [0, 0, 0]
];
exports.ResourceNotFoundException$ = ResourceNotFoundException$;
n0_registry.registerError(ResourceNotFoundException$, ResourceNotFoundException);
const ResourceNotReadyException$ = [-3, n0, _RNRE,
    { [_e]: _c },
    [_M, _C, _L],
    [0, 0, 0]
];
exports.ResourceNotReadyException$ = ResourceNotReadyException$;
n0_registry.registerError(ResourceNotReadyException$, ResourceNotReadyException);
const ServiceQuotaExceededException$ = [-3, n0, _SQEE,
    { [_e]: _c },
    [_M, _C, _L],
    [0, 0, 0]
];
exports.ServiceQuotaExceededException$ = ServiceQuotaExceededException$;
n0_registry.registerError(ServiceQuotaExceededException$, ServiceQuotaExceededException);
const SessionNotFoundException$ = [-3, n0, _SNFE,
    { [_e]: _c },
    [_M, _C, _L],
    [0, 0, 0]
];
exports.SessionNotFoundException$ = SessionNotFoundException$;
n0_registry.registerError(SessionNotFoundException$, SessionNotFoundException);
const ThrottlingException$ = [-3, n0, _TE,
    { [_e]: _se },
    [_M, _C, _L],
    [0, 0, 0]
];
exports.ThrottlingException$ = ThrottlingException$;
n0_registry.registerError(ThrottlingException$, ThrottlingException);
const VideoTooLargeException$ = [-3, n0, _VTLE,
    { [_e]: _c },
    [_M, _C, _L],
    [0, 0, 0]
];
exports.VideoTooLargeException$ = VideoTooLargeException$;
n0_registry.registerError(VideoTooLargeException$, VideoTooLargeException);
exports.errorTypeRegistries = [
    _s_registry,
    n0_registry,
];
var LivenessImageBlob = [0, n0, _LIB, 8, 21];
var SynthesizedJsonHumanLoopActivationConditionsEvaluationResults = [0, n0, _SJHLACER, { [_mT]: _a }, 0];
const AgeRange$ = [3, n0, _AR,
    0,
    [_Lo, _H],
    [1, 1]
];
exports.AgeRange$ = AgeRange$;
const Asset$ = [3, n0, _A,
    0,
    [_GTM],
    [() => GroundTruthManifest$]
];
exports.Asset$ = Asset$;
const AssociatedFace$ = [3, n0, _AF,
    0,
    [_FI],
    [0]
];
exports.AssociatedFace$ = AssociatedFace$;
const AssociateFacesRequest$ = [3, n0, _AFR,
    0,
    [_CI, _UI, _FIa, _UMT, _CRT],
    [0, 0, 64 | 0, 1, [0, 4]], 3
];
exports.AssociateFacesRequest$ = AssociateFacesRequest$;
const AssociateFacesResponse$ = [3, n0, _AFRs,
    0,
    [_AFs, _UFA, _US],
    [() => AssociatedFacesList, () => UnsuccessfulFaceAssociationList, 0]
];
exports.AssociateFacesResponse$ = AssociateFacesResponse$;
const AudioMetadata$ = [3, n0, _AM,
    0,
    [_Co, _DM, _SR, _NOC],
    [0, 1, 1, 1]
];
exports.AudioMetadata$ = AudioMetadata$;
const AuditImage$ = [3, n0, _AI,
    0,
    [_B, _SO, _BB],
    [[() => LivenessImageBlob, 0], () => S3Object$, () => BoundingBox$]
];
exports.AuditImage$ = AuditImage$;
const Beard$ = [3, n0, _Be,
    0,
    [_V, _Con],
    [2, 1]
];
exports.Beard$ = Beard$;
const BlackFrame$ = [3, n0, _BF,
    0,
    [_MPT, _MCP],
    [1, 1]
];
exports.BlackFrame$ = BlackFrame$;
const BoundingBox$ = [3, n0, _BB,
    0,
    [_W, _He, _Le, _T],
    [1, 1, 1, 1]
];
exports.BoundingBox$ = BoundingBox$;
const Celebrity$ = [3, n0, _Ce,
    0,
    [_U, _N, _I, _F, _MC, _KG],
    [64 | 0, 0, 0, () => ComparedFace$, 1, () => KnownGender$]
];
exports.Celebrity$ = Celebrity$;
const CelebrityDetail$ = [3, n0, _CD,
    0,
    [_U, _N, _I, _Con, _BB, _F, _KG],
    [64 | 0, 0, 0, 1, () => BoundingBox$, () => FaceDetail$, () => KnownGender$]
];
exports.CelebrityDetail$ = CelebrityDetail$;
const CelebrityRecognition$ = [3, n0, _CR,
    0,
    [_Ti, _Ce],
    [1, () => CelebrityDetail$]
];
exports.CelebrityRecognition$ = CelebrityRecognition$;
const Challenge$ = [3, n0, _Ch,
    0,
    [_Ty, _Ve],
    [0, 0], 2
];
exports.Challenge$ = Challenge$;
const ChallengePreference$ = [3, n0, _CP,
    0,
    [_Ty, _Ver],
    [0, () => Versions$], 1
];
exports.ChallengePreference$ = ChallengePreference$;
const ComparedFace$ = [3, n0, _CF,
    0,
    [_BB, _Con, _La, _P, _Q, _E, _S],
    [() => BoundingBox$, 1, () => Landmarks, () => Pose$, () => ImageQuality$, () => Emotions, () => Smile$]
];
exports.ComparedFace$ = ComparedFace$;
const ComparedSourceImageFace$ = [3, n0, _CSIF,
    0,
    [_BB, _Con],
    [() => BoundingBox$, 1]
];
exports.ComparedSourceImageFace$ = ComparedSourceImageFace$;
const CompareFacesMatch$ = [3, n0, _CFM,
    0,
    [_Si, _F],
    [1, () => ComparedFace$]
];
exports.CompareFacesMatch$ = CompareFacesMatch$;
const CompareFacesRequest$ = [3, n0, _CFR,
    0,
    [_SI, _TI, _ST, _QF],
    [() => Image$, () => Image$, 1, 0], 2
];
exports.CompareFacesRequest$ = CompareFacesRequest$;
const CompareFacesResponse$ = [3, n0, _CFRo,
    0,
    [_SIF, _FM, _UF, _SIOC, _TIOC],
    [() => ComparedSourceImageFace$, () => CompareFacesMatchList, () => CompareFacesUnmatchList, 0, 0]
];
exports.CompareFacesResponse$ = CompareFacesResponse$;
const ConnectedHomeSettings$ = [3, n0, _CHS,
    0,
    [_Lab, _MCi],
    [64 | 0, 1], 1
];
exports.ConnectedHomeSettings$ = ConnectedHomeSettings$;
const ConnectedHomeSettingsForUpdate$ = [3, n0, _CHSFU,
    0,
    [_Lab, _MCi],
    [64 | 0, 1]
];
exports.ConnectedHomeSettingsForUpdate$ = ConnectedHomeSettingsForUpdate$;
const ContentModerationDetection$ = [3, n0, _CMD,
    0,
    [_Ti, _ML, _STM, _ETM, _DM, _CT],
    [1, () => ModerationLabel$, 1, 1, 1, () => ContentTypes]
];
exports.ContentModerationDetection$ = ContentModerationDetection$;
const ContentType$ = [3, n0, _CTo,
    0,
    [_Con, _N],
    [1, 0]
];
exports.ContentType$ = ContentType$;
const CopyProjectVersionRequest$ = [3, n0, _CPVR,
    0,
    [_SPA, _SPVA, _DPA, _VN, _OC, _Ta, _KKI],
    [0, 0, 0, 0, () => OutputConfig$, 128 | 0, 0], 5
];
exports.CopyProjectVersionRequest$ = CopyProjectVersionRequest$;
const CopyProjectVersionResponse$ = [3, n0, _CPVRo,
    0,
    [_PVA],
    [0]
];
exports.CopyProjectVersionResponse$ = CopyProjectVersionResponse$;
const CoversBodyPart$ = [3, n0, _CBP,
    0,
    [_Con, _V],
    [1, 2]
];
exports.CoversBodyPart$ = CoversBodyPart$;
const CreateCollectionRequest$ = [3, n0, _CCR,
    0,
    [_CI, _Ta],
    [0, 128 | 0], 1
];
exports.CreateCollectionRequest$ = CreateCollectionRequest$;
const CreateCollectionResponse$ = [3, n0, _CCRr,
    0,
    [_SCt, _CA, _FMV],
    [1, 0, 0]
];
exports.CreateCollectionResponse$ = CreateCollectionResponse$;
const CreateDatasetRequest$ = [3, n0, _CDR,
    0,
    [_DT, _PA, _DS, _Ta],
    [0, 0, () => DatasetSource$, 128 | 0], 2
];
exports.CreateDatasetRequest$ = CreateDatasetRequest$;
const CreateDatasetResponse$ = [3, n0, _CDRr,
    0,
    [_DA],
    [0]
];
exports.CreateDatasetResponse$ = CreateDatasetResponse$;
const CreateFaceLivenessSessionRequest$ = [3, n0, _CFLSR,
    0,
    [_KKI, _Se, _CRT],
    [0, () => CreateFaceLivenessSessionRequestSettings$, 0]
];
exports.CreateFaceLivenessSessionRequest$ = CreateFaceLivenessSessionRequest$;
const CreateFaceLivenessSessionRequestSettings$ = [3, n0, _CFLSRS,
    0,
    [_OC, _AIL, _CPh],
    [() => LivenessOutputConfig$, 1, () => ChallengePreferences]
];
exports.CreateFaceLivenessSessionRequestSettings$ = CreateFaceLivenessSessionRequestSettings$;
const CreateFaceLivenessSessionResponse$ = [3, n0, _CFLSRr,
    0,
    [_SIe],
    [0], 1
];
exports.CreateFaceLivenessSessionResponse$ = CreateFaceLivenessSessionResponse$;
const CreateProjectRequest$ = [3, n0, _CPR,
    0,
    [_PN, _Fe, _AU, _Ta],
    [0, 0, 0, 128 | 0], 1
];
exports.CreateProjectRequest$ = CreateProjectRequest$;
const CreateProjectResponse$ = [3, n0, _CPRr,
    0,
    [_PA],
    [0]
];
exports.CreateProjectResponse$ = CreateProjectResponse$;
const CreateProjectVersionRequest$ = [3, n0, _CPVRr,
    0,
    [_PA, _VN, _OC, _TD, _TDe, _Ta, _KKI, _VD, _FC],
    [0, 0, () => OutputConfig$, () => TrainingData$, () => TestingData$, 128 | 0, 0, 0, () => CustomizationFeatureConfig$], 3
];
exports.CreateProjectVersionRequest$ = CreateProjectVersionRequest$;
const CreateProjectVersionResponse$ = [3, n0, _CPVRre,
    0,
    [_PVA],
    [0]
];
exports.CreateProjectVersionResponse$ = CreateProjectVersionResponse$;
const CreateStreamProcessorRequest$ = [3, n0, _CSPR,
    0,
    [_In, _O, _N, _Se, _RA, _Ta, _NC, _KKI, _ROI, _DSP],
    [() => StreamProcessorInput$, () => StreamProcessorOutput$, 0, () => StreamProcessorSettings$, 0, 128 | 0, () => StreamProcessorNotificationChannel$, 0, () => RegionsOfInterest, () => StreamProcessorDataSharingPreference$], 5
];
exports.CreateStreamProcessorRequest$ = CreateStreamProcessorRequest$;
const CreateStreamProcessorResponse$ = [3, n0, _CSPRr,
    0,
    [_SPAt],
    [0]
];
exports.CreateStreamProcessorResponse$ = CreateStreamProcessorResponse$;
const CreateUserRequest$ = [3, n0, _CUR,
    0,
    [_CI, _UI, _CRT],
    [0, 0, [0, 4]], 2
];
exports.CreateUserRequest$ = CreateUserRequest$;
const CreateUserResponse$ = [3, n0, _CURr,
    0,
    [],
    []
];
exports.CreateUserResponse$ = CreateUserResponse$;
const CustomizationFeatureConfig$ = [3, n0, _CFC,
    0,
    [_CM],
    [() => CustomizationFeatureContentModerationConfig$]
];
exports.CustomizationFeatureConfig$ = CustomizationFeatureConfig$;
const CustomizationFeatureContentModerationConfig$ = [3, n0, _CFCMC,
    0,
    [_CTon],
    [1]
];
exports.CustomizationFeatureContentModerationConfig$ = CustomizationFeatureContentModerationConfig$;
const CustomLabel$ = [3, n0, _CL,
    0,
    [_N, _Con, _G],
    [0, 1, () => Geometry$]
];
exports.CustomLabel$ = CustomLabel$;
const DatasetChanges$ = [3, n0, _DC,
    0,
    [_GT],
    [21], 1
];
exports.DatasetChanges$ = DatasetChanges$;
const DatasetDescription$ = [3, n0, _DD,
    0,
    [_CTr, _LUT, _St, _SM, _SMC, _DSa],
    [4, 4, 0, 0, 0, () => DatasetStats$]
];
exports.DatasetDescription$ = DatasetDescription$;
const DatasetLabelDescription$ = [3, n0, _DLD,
    0,
    [_LN, _LS],
    [0, () => DatasetLabelStats$]
];
exports.DatasetLabelDescription$ = DatasetLabelDescription$;
const DatasetLabelStats$ = [3, n0, _DLS,
    0,
    [_EC, _BBC],
    [1, 1]
];
exports.DatasetLabelStats$ = DatasetLabelStats$;
const DatasetMetadata$ = [3, n0, _DMa,
    0,
    [_CTr, _DT, _DA, _St, _SM, _SMC],
    [4, 0, 0, 0, 0, 0]
];
exports.DatasetMetadata$ = DatasetMetadata$;
const DatasetSource$ = [3, n0, _DS,
    0,
    [_GTM, _DA],
    [() => GroundTruthManifest$, 0]
];
exports.DatasetSource$ = DatasetSource$;
const DatasetStats$ = [3, n0, _DSa,
    0,
    [_LE, _TEo, _TL, _EE],
    [1, 1, 1, 1]
];
exports.DatasetStats$ = DatasetStats$;
const DeleteCollectionRequest$ = [3, n0, _DCR,
    0,
    [_CI],
    [0], 1
];
exports.DeleteCollectionRequest$ = DeleteCollectionRequest$;
const DeleteCollectionResponse$ = [3, n0, _DCRe,
    0,
    [_SCt],
    [1]
];
exports.DeleteCollectionResponse$ = DeleteCollectionResponse$;
const DeleteDatasetRequest$ = [3, n0, _DDR,
    0,
    [_DA],
    [0], 1
];
exports.DeleteDatasetRequest$ = DeleteDatasetRequest$;
const DeleteDatasetResponse$ = [3, n0, _DDRe,
    0,
    [],
    []
];
exports.DeleteDatasetResponse$ = DeleteDatasetResponse$;
const DeleteFacesRequest$ = [3, n0, _DFR,
    0,
    [_CI, _FIa],
    [0, 64 | 0], 2
];
exports.DeleteFacesRequest$ = DeleteFacesRequest$;
const DeleteFacesResponse$ = [3, n0, _DFRe,
    0,
    [_DF, _UFD],
    [64 | 0, () => UnsuccessfulFaceDeletionsList]
];
exports.DeleteFacesResponse$ = DeleteFacesResponse$;
const DeleteProjectPolicyRequest$ = [3, n0, _DPPR,
    0,
    [_PA, _PNo, _PRI],
    [0, 0, 0], 2
];
exports.DeleteProjectPolicyRequest$ = DeleteProjectPolicyRequest$;
const DeleteProjectPolicyResponse$ = [3, n0, _DPPRe,
    0,
    [],
    []
];
exports.DeleteProjectPolicyResponse$ = DeleteProjectPolicyResponse$;
const DeleteProjectRequest$ = [3, n0, _DPR,
    0,
    [_PA],
    [0], 1
];
exports.DeleteProjectRequest$ = DeleteProjectRequest$;
const DeleteProjectResponse$ = [3, n0, _DPRe,
    0,
    [_St],
    [0]
];
exports.DeleteProjectResponse$ = DeleteProjectResponse$;
const DeleteProjectVersionRequest$ = [3, n0, _DPVR,
    0,
    [_PVA],
    [0], 1
];
exports.DeleteProjectVersionRequest$ = DeleteProjectVersionRequest$;
const DeleteProjectVersionResponse$ = [3, n0, _DPVRe,
    0,
    [_St],
    [0]
];
exports.DeleteProjectVersionResponse$ = DeleteProjectVersionResponse$;
const DeleteStreamProcessorRequest$ = [3, n0, _DSPR,
    0,
    [_N],
    [0], 1
];
exports.DeleteStreamProcessorRequest$ = DeleteStreamProcessorRequest$;
const DeleteStreamProcessorResponse$ = [3, n0, _DSPRe,
    0,
    [],
    []
];
exports.DeleteStreamProcessorResponse$ = DeleteStreamProcessorResponse$;
const DeleteUserRequest$ = [3, n0, _DUR,
    0,
    [_CI, _UI, _CRT],
    [0, 0, [0, 4]], 2
];
exports.DeleteUserRequest$ = DeleteUserRequest$;
const DeleteUserResponse$ = [3, n0, _DURe,
    0,
    [],
    []
];
exports.DeleteUserResponse$ = DeleteUserResponse$;
const DescribeCollectionRequest$ = [3, n0, _DCRes,
    0,
    [_CI],
    [0], 1
];
exports.DescribeCollectionRequest$ = DescribeCollectionRequest$;
const DescribeCollectionResponse$ = [3, n0, _DCResc,
    0,
    [_FCa, _FMV, _CARN, _CTr, _UC],
    [1, 0, 0, 4, 1]
];
exports.DescribeCollectionResponse$ = DescribeCollectionResponse$;
const DescribeDatasetRequest$ = [3, n0, _DDRes,
    0,
    [_DA],
    [0], 1
];
exports.DescribeDatasetRequest$ = DescribeDatasetRequest$;
const DescribeDatasetResponse$ = [3, n0, _DDResc,
    0,
    [_DD],
    [() => DatasetDescription$]
];
exports.DescribeDatasetResponse$ = DescribeDatasetResponse$;
const DescribeProjectsRequest$ = [3, n0, _DPRes,
    0,
    [_NT, _MR, _PNr, _Fea],
    [0, 1, 64 | 0, 64 | 0]
];
exports.DescribeProjectsRequest$ = DescribeProjectsRequest$;
const DescribeProjectsResponse$ = [3, n0, _DPResc,
    0,
    [_PD, _NT],
    [() => ProjectDescriptions, 0]
];
exports.DescribeProjectsResponse$ = DescribeProjectsResponse$;
const DescribeProjectVersionsRequest$ = [3, n0, _DPVRes,
    0,
    [_PA, _VNe, _NT, _MR],
    [0, 64 | 0, 0, 1], 1
];
exports.DescribeProjectVersionsRequest$ = DescribeProjectVersionsRequest$;
const DescribeProjectVersionsResponse$ = [3, n0, _DPVResc,
    0,
    [_PVD, _NT],
    [() => ProjectVersionDescriptions, 0]
];
exports.DescribeProjectVersionsResponse$ = DescribeProjectVersionsResponse$;
const DescribeStreamProcessorRequest$ = [3, n0, _DSPRes,
    0,
    [_N],
    [0], 1
];
exports.DescribeStreamProcessorRequest$ = DescribeStreamProcessorRequest$;
const DescribeStreamProcessorResponse$ = [3, n0, _DSPResc,
    0,
    [_N, _SPAt, _St, _SM, _CTr, _LUTa, _In, _O, _RA, _Se, _NC, _KKI, _ROI, _DSP],
    [0, 0, 0, 0, 4, 4, () => StreamProcessorInput$, () => StreamProcessorOutput$, 0, () => StreamProcessorSettings$, () => StreamProcessorNotificationChannel$, 0, () => RegionsOfInterest, () => StreamProcessorDataSharingPreference$]
];
exports.DescribeStreamProcessorResponse$ = DescribeStreamProcessorResponse$;
const DetectCustomLabelsRequest$ = [3, n0, _DCLR,
    0,
    [_PVA, _Im, _MR, _MCi],
    [0, () => Image$, 1, 1], 2
];
exports.DetectCustomLabelsRequest$ = DetectCustomLabelsRequest$;
const DetectCustomLabelsResponse$ = [3, n0, _DCLRe,
    0,
    [_CLu],
    [() => CustomLabels]
];
exports.DetectCustomLabelsResponse$ = DetectCustomLabelsResponse$;
const DetectFacesRequest$ = [3, n0, _DFRet,
    0,
    [_Im, _At],
    [() => Image$, 64 | 0], 1
];
exports.DetectFacesRequest$ = DetectFacesRequest$;
const DetectFacesResponse$ = [3, n0, _DFRete,
    0,
    [_FD, _OCr],
    [() => FaceDetailList, 0]
];
exports.DetectFacesResponse$ = DetectFacesResponse$;
const DetectionFilter$ = [3, n0, _DFe,
    0,
    [_MCi, _MBBH, _MBBW],
    [1, 1, 1]
];
exports.DetectionFilter$ = DetectionFilter$;
const DetectLabelsImageBackground$ = [3, n0, _DLIB,
    0,
    [_Q, _DCo],
    [() => DetectLabelsImageQuality$, () => DominantColors]
];
exports.DetectLabelsImageBackground$ = DetectLabelsImageBackground$;
const DetectLabelsImageForeground$ = [3, n0, _DLIF,
    0,
    [_Q, _DCo],
    [() => DetectLabelsImageQuality$, () => DominantColors]
];
exports.DetectLabelsImageForeground$ = DetectLabelsImageForeground$;
const DetectLabelsImageProperties$ = [3, n0, _DLIP,
    0,
    [_Q, _DCo, _Fo, _Ba],
    [() => DetectLabelsImageQuality$, () => DominantColors, () => DetectLabelsImageForeground$, () => DetectLabelsImageBackground$]
];
exports.DetectLabelsImageProperties$ = DetectLabelsImageProperties$;
const DetectLabelsImagePropertiesSettings$ = [3, n0, _DLIPS,
    0,
    [_MDC],
    [1]
];
exports.DetectLabelsImagePropertiesSettings$ = DetectLabelsImagePropertiesSettings$;
const DetectLabelsImageQuality$ = [3, n0, _DLIQ,
    0,
    [_Br, _Sh, _Cont],
    [1, 1, 1]
];
exports.DetectLabelsImageQuality$ = DetectLabelsImageQuality$;
const DetectLabelsRequest$ = [3, n0, _DLR,
    0,
    [_Im, _MLa, _MCi, _Fea, _Se],
    [() => Image$, 1, 1, 64 | 0, () => DetectLabelsSettings$], 1
];
exports.DetectLabelsRequest$ = DetectLabelsRequest$;
const DetectLabelsResponse$ = [3, n0, _DLRe,
    0,
    [_Lab, _OCr, _LMV, _IP],
    [() => Labels, 0, 0, () => DetectLabelsImageProperties$]
];
exports.DetectLabelsResponse$ = DetectLabelsResponse$;
const DetectLabelsSettings$ = [3, n0, _DLSe,
    0,
    [_GL, _IP],
    [() => GeneralLabelsSettings$, () => DetectLabelsImagePropertiesSettings$]
];
exports.DetectLabelsSettings$ = DetectLabelsSettings$;
const DetectModerationLabelsRequest$ = [3, n0, _DMLR,
    0,
    [_Im, _MCi, _HLC, _PV],
    [() => Image$, 1, () => HumanLoopConfig$, 0], 1
];
exports.DetectModerationLabelsRequest$ = DetectModerationLabelsRequest$;
const DetectModerationLabelsResponse$ = [3, n0, _DMLRe,
    0,
    [_MLo, _MMV, _HLAO, _PV, _CT],
    [() => ModerationLabels, 0, [() => HumanLoopActivationOutput$, 0], 0, () => ContentTypes]
];
exports.DetectModerationLabelsResponse$ = DetectModerationLabelsResponse$;
const DetectProtectiveEquipmentRequest$ = [3, n0, _DPER,
    0,
    [_Im, _SA],
    [() => Image$, () => ProtectiveEquipmentSummarizationAttributes$], 1
];
exports.DetectProtectiveEquipmentRequest$ = DetectProtectiveEquipmentRequest$;
const DetectProtectiveEquipmentResponse$ = [3, n0, _DPERe,
    0,
    [_PEMV, _Pe, _Su],
    [0, () => ProtectiveEquipmentPersons, () => ProtectiveEquipmentSummary$]
];
exports.DetectProtectiveEquipmentResponse$ = DetectProtectiveEquipmentResponse$;
const DetectTextFilters$ = [3, n0, _DTF,
    0,
    [_WF, _ROI],
    [() => DetectionFilter$, () => RegionsOfInterest]
];
exports.DetectTextFilters$ = DetectTextFilters$;
const DetectTextRequest$ = [3, n0, _DTR,
    0,
    [_Im, _Fi],
    [() => Image$, () => DetectTextFilters$], 1
];
exports.DetectTextRequest$ = DetectTextRequest$;
const DetectTextResponse$ = [3, n0, _DTRe,
    0,
    [_TDex, _TMV],
    [() => TextDetectionList, 0]
];
exports.DetectTextResponse$ = DetectTextResponse$;
const DisassociatedFace$ = [3, n0, _DFi,
    0,
    [_FI],
    [0]
];
exports.DisassociatedFace$ = DisassociatedFace$;
const DisassociateFacesRequest$ = [3, n0, _DFRi,
    0,
    [_CI, _UI, _FIa, _CRT],
    [0, 0, 64 | 0, [0, 4]], 3
];
exports.DisassociateFacesRequest$ = DisassociateFacesRequest$;
const DisassociateFacesResponse$ = [3, n0, _DFRis,
    0,
    [_DFis, _UFDn, _US],
    [() => DisassociatedFacesList, () => UnsuccessfulFaceDisassociationList, 0]
];
exports.DisassociateFacesResponse$ = DisassociateFacesResponse$;
const DistributeDataset$ = [3, n0, _DDi,
    0,
    [_Ar],
    [0], 1
];
exports.DistributeDataset$ = DistributeDataset$;
const DistributeDatasetEntriesRequest$ = [3, n0, _DDER,
    0,
    [_D],
    [() => DistributeDatasetMetadataList], 1
];
exports.DistributeDatasetEntriesRequest$ = DistributeDatasetEntriesRequest$;
const DistributeDatasetEntriesResponse$ = [3, n0, _DDERi,
    0,
    [],
    []
];
exports.DistributeDatasetEntriesResponse$ = DistributeDatasetEntriesResponse$;
const DominantColor$ = [3, n0, _DCom,
    0,
    [_R, _Bl, _Gr, _HC, _CSSC, _SCi, _PP],
    [1, 1, 1, 0, 0, 0, 1]
];
exports.DominantColor$ = DominantColor$;
const Emotion$ = [3, n0, _Em,
    0,
    [_Ty, _Con],
    [0, 1]
];
exports.Emotion$ = Emotion$;
const EquipmentDetection$ = [3, n0, _ED,
    0,
    [_BB, _Con, _Ty, _CBP],
    [() => BoundingBox$, 1, 0, () => CoversBodyPart$]
];
exports.EquipmentDetection$ = EquipmentDetection$;
const EvaluationResult$ = [3, n0, _ER,
    0,
    [_FS, _Su],
    [1, () => Summary$]
];
exports.EvaluationResult$ = EvaluationResult$;
const EyeDirection$ = [3, n0, _EDy,
    0,
    [_Y, _Pi, _Con],
    [1, 1, 1]
];
exports.EyeDirection$ = EyeDirection$;
const Eyeglasses$ = [3, n0, _Ey,
    0,
    [_V, _Con],
    [2, 1]
];
exports.Eyeglasses$ = Eyeglasses$;
const EyeOpen$ = [3, n0, _EO,
    0,
    [_V, _Con],
    [2, 1]
];
exports.EyeOpen$ = EyeOpen$;
const Face$ = [3, n0, _F,
    0,
    [_FI, _BB, _II, _EII, _Con, _IFMV, _UI],
    [0, () => BoundingBox$, 0, 0, 1, 0, 0]
];
exports.Face$ = Face$;
const FaceDetail$ = [3, n0, _FDa,
    0,
    [_BB, _AR, _S, _Ey, _Sun, _Ge, _Be, _Mu, _EOy, _MO, _E, _La, _P, _Q, _Con, _FO, _EDy],
    [() => BoundingBox$, () => AgeRange$, () => Smile$, () => Eyeglasses$, () => Sunglasses$, () => Gender$, () => Beard$, () => Mustache$, () => EyeOpen$, () => MouthOpen$, () => Emotions, () => Landmarks, () => Pose$, () => ImageQuality$, 1, () => FaceOccluded$, () => EyeDirection$]
];
exports.FaceDetail$ = FaceDetail$;
const FaceDetection$ = [3, n0, _FDac,
    0,
    [_Ti, _F],
    [1, () => FaceDetail$]
];
exports.FaceDetection$ = FaceDetection$;
const FaceMatch$ = [3, n0, _FMa,
    0,
    [_Si, _F],
    [1, () => Face$]
];
exports.FaceMatch$ = FaceMatch$;
const FaceOccluded$ = [3, n0, _FO,
    0,
    [_V, _Con],
    [2, 1]
];
exports.FaceOccluded$ = FaceOccluded$;
const FaceRecord$ = [3, n0, _FR,
    0,
    [_F, _FDa],
    [() => Face$, () => FaceDetail$]
];
exports.FaceRecord$ = FaceRecord$;
const FaceSearchSettings$ = [3, n0, _FSS,
    0,
    [_CI, _FMT],
    [0, 1]
];
exports.FaceSearchSettings$ = FaceSearchSettings$;
const Gender$ = [3, n0, _Ge,
    0,
    [_V, _Con],
    [0, 1]
];
exports.Gender$ = Gender$;
const GeneralLabelsSettings$ = [3, n0, _GLS,
    0,
    [_LIF, _LEF, _LCIF, _LCEF],
    [64 | 0, 64 | 0, 64 | 0, 64 | 0]
];
exports.GeneralLabelsSettings$ = GeneralLabelsSettings$;
const Geometry$ = [3, n0, _G,
    0,
    [_BB, _Po],
    [() => BoundingBox$, () => Polygon]
];
exports.Geometry$ = Geometry$;
const GetCelebrityInfoRequest$ = [3, n0, _GCIR,
    0,
    [_I],
    [0], 1
];
exports.GetCelebrityInfoRequest$ = GetCelebrityInfoRequest$;
const GetCelebrityInfoResponse$ = [3, n0, _GCIRe,
    0,
    [_U, _N, _KG],
    [64 | 0, 0, () => KnownGender$]
];
exports.GetCelebrityInfoResponse$ = GetCelebrityInfoResponse$;
const GetCelebrityRecognitionRequest$ = [3, n0, _GCRR,
    0,
    [_JI, _MR, _NT, _SB],
    [0, 1, 0, 0], 1
];
exports.GetCelebrityRecognitionRequest$ = GetCelebrityRecognitionRequest$;
const GetCelebrityRecognitionResponse$ = [3, n0, _GCRRe,
    0,
    [_JS, _SM, _VM, _NT, _Cel, _JI, _Vi, _JT],
    [0, 0, () => VideoMetadata$, 0, () => CelebrityRecognitions, 0, () => Video$, 0]
];
exports.GetCelebrityRecognitionResponse$ = GetCelebrityRecognitionResponse$;
const GetContentModerationRequest$ = [3, n0, _GCMR,
    0,
    [_JI, _MR, _NT, _SB, _AB],
    [0, 1, 0, 0, 0], 1
];
exports.GetContentModerationRequest$ = GetContentModerationRequest$;
const GetContentModerationRequestMetadata$ = [3, n0, _GCMRM,
    0,
    [_SB, _AB],
    [0, 0]
];
exports.GetContentModerationRequestMetadata$ = GetContentModerationRequestMetadata$;
const GetContentModerationResponse$ = [3, n0, _GCMRe,
    0,
    [_JS, _SM, _VM, _MLo, _NT, _MMV, _JI, _Vi, _JT, _GRM],
    [0, 0, () => VideoMetadata$, () => ContentModerationDetections, 0, 0, 0, () => Video$, 0, () => GetContentModerationRequestMetadata$]
];
exports.GetContentModerationResponse$ = GetContentModerationResponse$;
const GetFaceDetectionRequest$ = [3, n0, _GFDR,
    0,
    [_JI, _MR, _NT],
    [0, 1, 0], 1
];
exports.GetFaceDetectionRequest$ = GetFaceDetectionRequest$;
const GetFaceDetectionResponse$ = [3, n0, _GFDRe,
    0,
    [_JS, _SM, _VM, _NT, _Fa, _JI, _Vi, _JT],
    [0, 0, () => VideoMetadata$, 0, () => FaceDetections, 0, () => Video$, 0]
];
exports.GetFaceDetectionResponse$ = GetFaceDetectionResponse$;
const GetFaceLivenessSessionResultsRequest$ = [3, n0, _GFLSRR,
    0,
    [_SIe],
    [0], 1
];
exports.GetFaceLivenessSessionResultsRequest$ = GetFaceLivenessSessionResultsRequest$;
const GetFaceLivenessSessionResultsResponse$ = [3, n0, _GFLSRRe,
    0,
    [_SIe, _St, _Con, _RI, _AIu, _Ch],
    [0, 0, 1, [() => AuditImage$, 0], [() => AuditImages, 0], () => Challenge$], 2
];
exports.GetFaceLivenessSessionResultsResponse$ = GetFaceLivenessSessionResultsResponse$;
const GetFaceSearchRequest$ = [3, n0, _GFSR,
    0,
    [_JI, _MR, _NT, _SB],
    [0, 1, 0, 0], 1
];
exports.GetFaceSearchRequest$ = GetFaceSearchRequest$;
const GetFaceSearchResponse$ = [3, n0, _GFSRe,
    0,
    [_JS, _SM, _NT, _VM, _Pe, _JI, _Vi, _JT],
    [0, 0, 0, () => VideoMetadata$, () => PersonMatches, 0, () => Video$, 0]
];
exports.GetFaceSearchResponse$ = GetFaceSearchResponse$;
const GetLabelDetectionRequest$ = [3, n0, _GLDR,
    0,
    [_JI, _MR, _NT, _SB, _AB],
    [0, 1, 0, 0, 0], 1
];
exports.GetLabelDetectionRequest$ = GetLabelDetectionRequest$;
const GetLabelDetectionRequestMetadata$ = [3, n0, _GLDRM,
    0,
    [_SB, _AB],
    [0, 0]
];
exports.GetLabelDetectionRequestMetadata$ = GetLabelDetectionRequestMetadata$;
const GetLabelDetectionResponse$ = [3, n0, _GLDRe,
    0,
    [_JS, _SM, _VM, _NT, _Lab, _LMV, _JI, _Vi, _JT, _GRM],
    [0, 0, () => VideoMetadata$, 0, () => LabelDetections, 0, 0, () => Video$, 0, () => GetLabelDetectionRequestMetadata$]
];
exports.GetLabelDetectionResponse$ = GetLabelDetectionResponse$;
const GetMediaAnalysisJobRequest$ = [3, n0, _GMAJR,
    0,
    [_JI],
    [0], 1
];
exports.GetMediaAnalysisJobRequest$ = GetMediaAnalysisJobRequest$;
const GetMediaAnalysisJobResponse$ = [3, n0, _GMAJRe,
    0,
    [_JI, _OCp, _St, _CTr, _In, _OC, _JN, _FDai, _CTom, _KKI, _Re, _MS],
    [0, () => MediaAnalysisOperationsConfig$, 0, 4, () => MediaAnalysisInput$, () => MediaAnalysisOutputConfig$, 0, () => MediaAnalysisJobFailureDetails$, 4, 0, () => MediaAnalysisResults$, () => MediaAnalysisManifestSummary$], 6
];
exports.GetMediaAnalysisJobResponse$ = GetMediaAnalysisJobResponse$;
const GetPersonTrackingRequest$ = [3, n0, _GPTR,
    0,
    [_JI, _MR, _NT, _SB],
    [0, 1, 0, 0], 1
];
exports.GetPersonTrackingRequest$ = GetPersonTrackingRequest$;
const GetPersonTrackingResponse$ = [3, n0, _GPTRe,
    0,
    [_JS, _SM, _VM, _NT, _Pe, _JI, _Vi, _JT],
    [0, 0, () => VideoMetadata$, 0, () => PersonDetections, 0, () => Video$, 0]
];
exports.GetPersonTrackingResponse$ = GetPersonTrackingResponse$;
const GetSegmentDetectionRequest$ = [3, n0, _GSDR,
    0,
    [_JI, _MR, _NT],
    [0, 1, 0], 1
];
exports.GetSegmentDetectionRequest$ = GetSegmentDetectionRequest$;
const GetSegmentDetectionResponse$ = [3, n0, _GSDRe,
    0,
    [_JS, _SM, _VM, _AM, _NT, _Seg, _SST, _JI, _Vi, _JT],
    [0, 0, () => VideoMetadataList, () => AudioMetadataList, 0, () => SegmentDetections, () => SegmentTypesInfo, 0, () => Video$, 0]
];
exports.GetSegmentDetectionResponse$ = GetSegmentDetectionResponse$;
const GetTextDetectionRequest$ = [3, n0, _GTDR,
    0,
    [_JI, _MR, _NT],
    [0, 1, 0], 1
];
exports.GetTextDetectionRequest$ = GetTextDetectionRequest$;
const GetTextDetectionResponse$ = [3, n0, _GTDRe,
    0,
    [_JS, _SM, _VM, _TDex, _NT, _TMV, _JI, _Vi, _JT],
    [0, 0, () => VideoMetadata$, () => TextDetectionResults, 0, 0, 0, () => Video$, 0]
];
exports.GetTextDetectionResponse$ = GetTextDetectionResponse$;
const GroundTruthManifest$ = [3, n0, _GTM,
    0,
    [_SO],
    [() => S3Object$]
];
exports.GroundTruthManifest$ = GroundTruthManifest$;
const HumanLoopActivationOutput$ = [3, n0, _HLAO,
    0,
    [_HLA, _HLAR, _HLACER],
    [0, 64 | 0, [() => SynthesizedJsonHumanLoopActivationConditionsEvaluationResults, 0]]
];
exports.HumanLoopActivationOutput$ = HumanLoopActivationOutput$;
const HumanLoopConfig$ = [3, n0, _HLC,
    0,
    [_HLN, _FDA, _DAa],
    [0, 0, () => HumanLoopDataAttributes$], 2
];
exports.HumanLoopConfig$ = HumanLoopConfig$;
const HumanLoopDataAttributes$ = [3, n0, _HLDA,
    0,
    [_CC],
    [64 | 0]
];
exports.HumanLoopDataAttributes$ = HumanLoopDataAttributes$;
const Image$ = [3, n0, _Im,
    0,
    [_B, _SO],
    [21, () => S3Object$]
];
exports.Image$ = Image$;
const ImageQuality$ = [3, n0, _IQ,
    0,
    [_Br, _Sh],
    [1, 1]
];
exports.ImageQuality$ = ImageQuality$;
const IndexFacesRequest$ = [3, n0, _IFR,
    0,
    [_CI, _Im, _EII, _DAe, _MF, _QF],
    [0, () => Image$, 0, 64 | 0, 1, 0], 2
];
exports.IndexFacesRequest$ = IndexFacesRequest$;
const IndexFacesResponse$ = [3, n0, _IFRn,
    0,
    [_FRa, _OCr, _FMV, _UFn],
    [() => FaceRecordList, 0, 0, () => UnindexedFaces]
];
exports.IndexFacesResponse$ = IndexFacesResponse$;
const Instance$ = [3, n0, _Ins,
    0,
    [_BB, _Con, _DCo],
    [() => BoundingBox$, 1, () => DominantColors]
];
exports.Instance$ = Instance$;
const KinesisDataStream$ = [3, n0, _KDS,
    0,
    [_Ar],
    [0]
];
exports.KinesisDataStream$ = KinesisDataStream$;
const KinesisVideoStream$ = [3, n0, _KVS,
    0,
    [_Ar],
    [0]
];
exports.KinesisVideoStream$ = KinesisVideoStream$;
const KinesisVideoStreamStartSelector$ = [3, n0, _KVSSS,
    0,
    [_PT, _FN],
    [1, 0]
];
exports.KinesisVideoStreamStartSelector$ = KinesisVideoStreamStartSelector$;
const KnownGender$ = [3, n0, _KG,
    0,
    [_Ty],
    [0]
];
exports.KnownGender$ = KnownGender$;
const Label$ = [3, n0, _Labe,
    0,
    [_N, _Con, _Inst, _Pa, _Al, _Ca],
    [0, 1, () => Instances, () => Parents, () => LabelAliases, () => LabelCategories]
];
exports.Label$ = Label$;
const LabelAlias$ = [3, n0, _LA,
    0,
    [_N],
    [0]
];
exports.LabelAlias$ = LabelAlias$;
const LabelCategory$ = [3, n0, _LC,
    0,
    [_N],
    [0]
];
exports.LabelCategory$ = LabelCategory$;
const LabelDetection$ = [3, n0, _LD,
    0,
    [_Ti, _Labe, _STM, _ETM, _DM],
    [1, () => Label$, 1, 1, 1]
];
exports.LabelDetection$ = LabelDetection$;
const LabelDetectionSettings$ = [3, n0, _LDS,
    0,
    [_GL],
    [() => GeneralLabelsSettings$]
];
exports.LabelDetectionSettings$ = LabelDetectionSettings$;
const Landmark$ = [3, n0, _Lan,
    0,
    [_Ty, _X, _Y_],
    [0, 1, 1]
];
exports.Landmark$ = Landmark$;
const ListCollectionsRequest$ = [3, n0, _LCR,
    0,
    [_NT, _MR],
    [0, 1]
];
exports.ListCollectionsRequest$ = ListCollectionsRequest$;
const ListCollectionsResponse$ = [3, n0, _LCRi,
    0,
    [_CIo, _NT, _FMVa],
    [64 | 0, 0, 64 | 0]
];
exports.ListCollectionsResponse$ = ListCollectionsResponse$;
const ListDatasetEntriesRequest$ = [3, n0, _LDER,
    0,
    [_DA, _CLo, _Label, _SRC, _HE, _NT, _MR],
    [0, 64 | 0, 2, 0, 2, 0, 1], 1
];
exports.ListDatasetEntriesRequest$ = ListDatasetEntriesRequest$;
const ListDatasetEntriesResponse$ = [3, n0, _LDERi,
    0,
    [_DE, _NT],
    [64 | 0, 0]
];
exports.ListDatasetEntriesResponse$ = ListDatasetEntriesResponse$;
const ListDatasetLabelsRequest$ = [3, n0, _LDLR,
    0,
    [_DA, _NT, _MR],
    [0, 0, 1], 1
];
exports.ListDatasetLabelsRequest$ = ListDatasetLabelsRequest$;
const ListDatasetLabelsResponse$ = [3, n0, _LDLRi,
    0,
    [_DLDa, _NT],
    [() => DatasetLabelDescriptions, 0]
];
exports.ListDatasetLabelsResponse$ = ListDatasetLabelsResponse$;
const ListFacesRequest$ = [3, n0, _LFR,
    0,
    [_CI, _NT, _MR, _UI, _FIa],
    [0, 0, 1, 0, 64 | 0], 1
];
exports.ListFacesRequest$ = ListFacesRequest$;
const ListFacesResponse$ = [3, n0, _LFRi,
    0,
    [_Fa, _NT, _FMV],
    [() => FaceList, 0, 0]
];
exports.ListFacesResponse$ = ListFacesResponse$;
const ListMediaAnalysisJobsRequest$ = [3, n0, _LMAJR,
    0,
    [_NT, _MR],
    [0, 1]
];
exports.ListMediaAnalysisJobsRequest$ = ListMediaAnalysisJobsRequest$;
const ListMediaAnalysisJobsResponse$ = [3, n0, _LMAJRi,
    0,
    [_MAJ, _NT],
    [() => MediaAnalysisJobDescriptions, 0], 1
];
exports.ListMediaAnalysisJobsResponse$ = ListMediaAnalysisJobsResponse$;
const ListProjectPoliciesRequest$ = [3, n0, _LPPR,
    0,
    [_PA, _NT, _MR],
    [0, 0, 1], 1
];
exports.ListProjectPoliciesRequest$ = ListProjectPoliciesRequest$;
const ListProjectPoliciesResponse$ = [3, n0, _LPPRi,
    0,
    [_PPr, _NT],
    [() => ProjectPolicies, 0]
];
exports.ListProjectPoliciesResponse$ = ListProjectPoliciesResponse$;
const ListStreamProcessorsRequest$ = [3, n0, _LSPR,
    0,
    [_NT, _MR],
    [0, 1]
];
exports.ListStreamProcessorsRequest$ = ListStreamProcessorsRequest$;
const ListStreamProcessorsResponse$ = [3, n0, _LSPRi,
    0,
    [_NT, _SP],
    [0, () => StreamProcessorList]
];
exports.ListStreamProcessorsResponse$ = ListStreamProcessorsResponse$;
const ListTagsForResourceRequest$ = [3, n0, _LTFRR,
    0,
    [_RAe],
    [0], 1
];
exports.ListTagsForResourceRequest$ = ListTagsForResourceRequest$;
const ListTagsForResourceResponse$ = [3, n0, _LTFRRi,
    0,
    [_Ta],
    [128 | 0]
];
exports.ListTagsForResourceResponse$ = ListTagsForResourceResponse$;
const ListUsersRequest$ = [3, n0, _LUR,
    0,
    [_CI, _MR, _NT],
    [0, 1, 0], 1
];
exports.ListUsersRequest$ = ListUsersRequest$;
const ListUsersResponse$ = [3, n0, _LURi,
    0,
    [_Us, _NT],
    [() => UserList, 0]
];
exports.ListUsersResponse$ = ListUsersResponse$;
const LivenessOutputConfig$ = [3, n0, _LOC,
    0,
    [_SBu, _SKP],
    [0, 0], 1
];
exports.LivenessOutputConfig$ = LivenessOutputConfig$;
const MatchedUser$ = [3, n0, _MU,
    0,
    [_UI, _US],
    [0, 0]
];
exports.MatchedUser$ = MatchedUser$;
const MediaAnalysisDetectModerationLabelsConfig$ = [3, n0, _MADMLC,
    0,
    [_MCi, _PV],
    [1, 0]
];
exports.MediaAnalysisDetectModerationLabelsConfig$ = MediaAnalysisDetectModerationLabelsConfig$;
const MediaAnalysisInput$ = [3, n0, _MAI,
    0,
    [_SO],
    [() => S3Object$], 1
];
exports.MediaAnalysisInput$ = MediaAnalysisInput$;
const MediaAnalysisJobDescription$ = [3, n0, _MAJD,
    0,
    [_JI, _OCp, _St, _CTr, _In, _OC, _JN, _FDai, _CTom, _KKI, _Re, _MS],
    [0, () => MediaAnalysisOperationsConfig$, 0, 4, () => MediaAnalysisInput$, () => MediaAnalysisOutputConfig$, 0, () => MediaAnalysisJobFailureDetails$, 4, 0, () => MediaAnalysisResults$, () => MediaAnalysisManifestSummary$], 6
];
exports.MediaAnalysisJobDescription$ = MediaAnalysisJobDescription$;
const MediaAnalysisJobFailureDetails$ = [3, n0, _MAJFD,
    0,
    [_C, _M],
    [0, 0]
];
exports.MediaAnalysisJobFailureDetails$ = MediaAnalysisJobFailureDetails$;
const MediaAnalysisManifestSummary$ = [3, n0, _MAMS,
    0,
    [_SO],
    [() => S3Object$]
];
exports.MediaAnalysisManifestSummary$ = MediaAnalysisManifestSummary$;
const MediaAnalysisModelVersions$ = [3, n0, _MAMV,
    0,
    [_Mo],
    [0]
];
exports.MediaAnalysisModelVersions$ = MediaAnalysisModelVersions$;
const MediaAnalysisOperationsConfig$ = [3, n0, _MAOC,
    0,
    [_DML],
    [() => MediaAnalysisDetectModerationLabelsConfig$]
];
exports.MediaAnalysisOperationsConfig$ = MediaAnalysisOperationsConfig$;
const MediaAnalysisOutputConfig$ = [3, n0, _MAOCe,
    0,
    [_SBu, _SKP],
    [0, 0], 1
];
exports.MediaAnalysisOutputConfig$ = MediaAnalysisOutputConfig$;
const MediaAnalysisResults$ = [3, n0, _MAR,
    0,
    [_SO, _MV],
    [() => S3Object$, () => MediaAnalysisModelVersions$]
];
exports.MediaAnalysisResults$ = MediaAnalysisResults$;
const ModerationLabel$ = [3, n0, _ML,
    0,
    [_Con, _N, _PNa, _TLa],
    [1, 0, 0, 1]
];
exports.ModerationLabel$ = ModerationLabel$;
const MouthOpen$ = [3, n0, _MO,
    0,
    [_V, _Con],
    [2, 1]
];
exports.MouthOpen$ = MouthOpen$;
const Mustache$ = [3, n0, _Mu,
    0,
    [_V, _Con],
    [2, 1]
];
exports.Mustache$ = Mustache$;
const NotificationChannel$ = [3, n0, _NC,
    0,
    [_SNSTA, _RA],
    [0, 0], 2
];
exports.NotificationChannel$ = NotificationChannel$;
const OutputConfig$ = [3, n0, _OC,
    0,
    [_SBu, _SKP],
    [0, 0]
];
exports.OutputConfig$ = OutputConfig$;
const Parent$ = [3, n0, _Par,
    0,
    [_N],
    [0]
];
exports.Parent$ = Parent$;
const PersonDetail$ = [3, n0, _PDe,
    0,
    [_Ind, _BB, _F],
    [1, () => BoundingBox$, () => FaceDetail$]
];
exports.PersonDetail$ = PersonDetail$;
const PersonDetection$ = [3, n0, _PDer,
    0,
    [_Ti, _Per],
    [1, () => PersonDetail$]
];
exports.PersonDetection$ = PersonDetection$;
const PersonMatch$ = [3, n0, _PM,
    0,
    [_Ti, _Per, _FM],
    [1, () => PersonDetail$, () => FaceMatchList]
];
exports.PersonMatch$ = PersonMatch$;
const Point$ = [3, n0, _Poi,
    0,
    [_X, _Y_],
    [1, 1]
];
exports.Point$ = Point$;
const Pose$ = [3, n0, _P,
    0,
    [_Ro, _Y, _Pi],
    [1, 1, 1]
];
exports.Pose$ = Pose$;
const ProjectDescription$ = [3, n0, _PDr,
    0,
    [_PA, _CTr, _St, _D, _Fe, _AU],
    [0, 4, 0, () => DatasetMetadataList, 0, 0]
];
exports.ProjectDescription$ = ProjectDescription$;
const ProjectPolicy$ = [3, n0, _PPro,
    0,
    [_PA, _PNo, _PRI, _PDo, _CTr, _LUT],
    [0, 0, 0, 0, 4, 4]
];
exports.ProjectPolicy$ = ProjectPolicy$;
const ProjectVersionDescription$ = [3, n0, _PVDr,
    0,
    [_PVA, _CTr, _MIU, _St, _SM, _BTTIS, _TET, _OC, _TDR, _TDRe, _ER, _MS, _KKI, _MIUa, _SPVA, _VD, _Fe, _BMV, _FC],
    [0, 4, 1, 0, 0, 1, 4, () => OutputConfig$, () => TrainingDataResult$, () => TestingDataResult$, () => EvaluationResult$, () => GroundTruthManifest$, 0, 1, 0, 0, 0, 0, () => CustomizationFeatureConfig$]
];
exports.ProjectVersionDescription$ = ProjectVersionDescription$;
const ProtectiveEquipmentBodyPart$ = [3, n0, _PEBP,
    0,
    [_N, _Con, _EDq],
    [0, 1, () => EquipmentDetections]
];
exports.ProtectiveEquipmentBodyPart$ = ProtectiveEquipmentBodyPart$;
const ProtectiveEquipmentPerson$ = [3, n0, _PEP,
    0,
    [_BP, _BB, _Con, _I],
    [() => BodyParts, () => BoundingBox$, 1, 1]
];
exports.ProtectiveEquipmentPerson$ = ProtectiveEquipmentPerson$;
const ProtectiveEquipmentSummarizationAttributes$ = [3, n0, _PESA,
    0,
    [_MCi, _RET],
    [1, 64 | 0], 2
];
exports.ProtectiveEquipmentSummarizationAttributes$ = ProtectiveEquipmentSummarizationAttributes$;
const ProtectiveEquipmentSummary$ = [3, n0, _PES,
    0,
    [_PWRE, _PWREe, _PI],
    [64 | 1, 64 | 1, 64 | 1]
];
exports.ProtectiveEquipmentSummary$ = ProtectiveEquipmentSummary$;
const PutProjectPolicyRequest$ = [3, n0, _PPPR,
    0,
    [_PA, _PNo, _PDo, _PRI],
    [0, 0, 0, 0], 3
];
exports.PutProjectPolicyRequest$ = PutProjectPolicyRequest$;
const PutProjectPolicyResponse$ = [3, n0, _PPPRu,
    0,
    [_PRI],
    [0]
];
exports.PutProjectPolicyResponse$ = PutProjectPolicyResponse$;
const RecognizeCelebritiesRequest$ = [3, n0, _RCR,
    0,
    [_Im],
    [() => Image$], 1
];
exports.RecognizeCelebritiesRequest$ = RecognizeCelebritiesRequest$;
const RecognizeCelebritiesResponse$ = [3, n0, _RCRe,
    0,
    [_CFe, _UFnr, _OCr],
    [() => CelebrityList, () => ComparedFaceList, 0]
];
exports.RecognizeCelebritiesResponse$ = RecognizeCelebritiesResponse$;
const RegionOfInterest$ = [3, n0, _ROIe,
    0,
    [_BB, _Po],
    [() => BoundingBox$, () => Polygon]
];
exports.RegionOfInterest$ = RegionOfInterest$;
const S3Destination$ = [3, n0, _SD,
    0,
    [_Bu, _KP],
    [0, 0]
];
exports.S3Destination$ = S3Destination$;
const S3Object$ = [3, n0, _SO,
    0,
    [_Bu, _N, _Ve],
    [0, 0, 0]
];
exports.S3Object$ = S3Object$;
const SearchedFace$ = [3, n0, _SF,
    0,
    [_FI],
    [0]
];
exports.SearchedFace$ = SearchedFace$;
const SearchedFaceDetails$ = [3, n0, _SFD,
    0,
    [_FDa],
    [() => FaceDetail$]
];
exports.SearchedFaceDetails$ = SearchedFaceDetails$;
const SearchedUser$ = [3, n0, _SU,
    0,
    [_UI],
    [0]
];
exports.SearchedUser$ = SearchedUser$;
const SearchFacesByImageRequest$ = [3, n0, _SFBIR,
    0,
    [_CI, _Im, _MF, _FMT, _QF],
    [0, () => Image$, 1, 1, 0], 2
];
exports.SearchFacesByImageRequest$ = SearchFacesByImageRequest$;
const SearchFacesByImageResponse$ = [3, n0, _SFBIRe,
    0,
    [_SFBB, _SFC, _FM, _FMV],
    [() => BoundingBox$, 1, () => FaceMatchList, 0]
];
exports.SearchFacesByImageResponse$ = SearchFacesByImageResponse$;
const SearchFacesRequest$ = [3, n0, _SFR,
    0,
    [_CI, _FI, _MF, _FMT],
    [0, 0, 1, 1], 2
];
exports.SearchFacesRequest$ = SearchFacesRequest$;
const SearchFacesResponse$ = [3, n0, _SFRe,
    0,
    [_SFI, _FM, _FMV],
    [0, () => FaceMatchList, 0]
];
exports.SearchFacesResponse$ = SearchFacesResponse$;
const SearchUsersByImageRequest$ = [3, n0, _SUBIR,
    0,
    [_CI, _Im, _UMT, _MUa, _QF],
    [0, () => Image$, 1, 1, 0], 2
];
exports.SearchUsersByImageRequest$ = SearchUsersByImageRequest$;
const SearchUsersByImageResponse$ = [3, n0, _SUBIRe,
    0,
    [_UM, _FMV, _SF, _UFns],
    [() => UserMatchList, 0, () => SearchedFaceDetails$, () => UnsearchedFacesList]
];
exports.SearchUsersByImageResponse$ = SearchUsersByImageResponse$;
const SearchUsersRequest$ = [3, n0, _SUR,
    0,
    [_CI, _UI, _FI, _UMT, _MUa],
    [0, 0, 0, 1, 1], 1
];
exports.SearchUsersRequest$ = SearchUsersRequest$;
const SearchUsersResponse$ = [3, n0, _SURe,
    0,
    [_UM, _FMV, _SF, _SU],
    [() => UserMatchList, 0, () => SearchedFace$, () => SearchedUser$]
];
exports.SearchUsersResponse$ = SearchUsersResponse$;
const SegmentDetection$ = [3, n0, _SDe,
    0,
    [_Ty, _STM, _ETM, _DM, _STSMPTE, _ETSMPTE, _DSMPTE, _TCS, _SS, _SFN, _EFN, _DFu],
    [0, 1, 1, 1, 0, 0, 0, () => TechnicalCueSegment$, () => ShotSegment$, 1, 1, 1]
];
exports.SegmentDetection$ = SegmentDetection$;
const SegmentTypeInfo$ = [3, n0, _STI,
    0,
    [_Ty, _MVo],
    [0, 0]
];
exports.SegmentTypeInfo$ = SegmentTypeInfo$;
const ShotSegment$ = [3, n0, _SS,
    0,
    [_Ind, _Con],
    [1, 1]
];
exports.ShotSegment$ = ShotSegment$;
const Smile$ = [3, n0, _S,
    0,
    [_V, _Con],
    [2, 1]
];
exports.Smile$ = Smile$;
const StartCelebrityRecognitionRequest$ = [3, n0, _SCRR,
    0,
    [_Vi, _CRT, _NC, _JT],
    [() => Video$, 0, () => NotificationChannel$, 0], 1
];
exports.StartCelebrityRecognitionRequest$ = StartCelebrityRecognitionRequest$;
const StartCelebrityRecognitionResponse$ = [3, n0, _SCRRt,
    0,
    [_JI],
    [0]
];
exports.StartCelebrityRecognitionResponse$ = StartCelebrityRecognitionResponse$;
const StartContentModerationRequest$ = [3, n0, _SCMR,
    0,
    [_Vi, _MCi, _CRT, _NC, _JT],
    [() => Video$, 1, 0, () => NotificationChannel$, 0], 1
];
exports.StartContentModerationRequest$ = StartContentModerationRequest$;
const StartContentModerationResponse$ = [3, n0, _SCMRt,
    0,
    [_JI],
    [0]
];
exports.StartContentModerationResponse$ = StartContentModerationResponse$;
const StartFaceDetectionRequest$ = [3, n0, _SFDR,
    0,
    [_Vi, _CRT, _NC, _FA, _JT],
    [() => Video$, 0, () => NotificationChannel$, 0, 0], 1
];
exports.StartFaceDetectionRequest$ = StartFaceDetectionRequest$;
const StartFaceDetectionResponse$ = [3, n0, _SFDRt,
    0,
    [_JI],
    [0]
];
exports.StartFaceDetectionResponse$ = StartFaceDetectionResponse$;
const StartFaceSearchRequest$ = [3, n0, _SFSR,
    0,
    [_Vi, _CI, _CRT, _FMT, _NC, _JT],
    [() => Video$, 0, 0, 1, () => NotificationChannel$, 0], 2
];
exports.StartFaceSearchRequest$ = StartFaceSearchRequest$;
const StartFaceSearchResponse$ = [3, n0, _SFSRt,
    0,
    [_JI],
    [0]
];
exports.StartFaceSearchResponse$ = StartFaceSearchResponse$;
const StartLabelDetectionRequest$ = [3, n0, _SLDR,
    0,
    [_Vi, _CRT, _MCi, _NC, _JT, _Fea, _Se],
    [() => Video$, 0, 1, () => NotificationChannel$, 0, 64 | 0, () => LabelDetectionSettings$], 1
];
exports.StartLabelDetectionRequest$ = StartLabelDetectionRequest$;
const StartLabelDetectionResponse$ = [3, n0, _SLDRt,
    0,
    [_JI],
    [0]
];
exports.StartLabelDetectionResponse$ = StartLabelDetectionResponse$;
const StartMediaAnalysisJobRequest$ = [3, n0, _SMAJR,
    0,
    [_OCp, _In, _OC, _CRT, _JN, _KKI],
    [() => MediaAnalysisOperationsConfig$, () => MediaAnalysisInput$, () => MediaAnalysisOutputConfig$, [0, 4], 0, 0], 3
];
exports.StartMediaAnalysisJobRequest$ = StartMediaAnalysisJobRequest$;
const StartMediaAnalysisJobResponse$ = [3, n0, _SMAJRt,
    0,
    [_JI],
    [0], 1
];
exports.StartMediaAnalysisJobResponse$ = StartMediaAnalysisJobResponse$;
const StartPersonTrackingRequest$ = [3, n0, _SPTR,
    0,
    [_Vi, _CRT, _NC, _JT],
    [() => Video$, 0, () => NotificationChannel$, 0], 1
];
exports.StartPersonTrackingRequest$ = StartPersonTrackingRequest$;
const StartPersonTrackingResponse$ = [3, n0, _SPTRt,
    0,
    [_JI],
    [0]
];
exports.StartPersonTrackingResponse$ = StartPersonTrackingResponse$;
const StartProjectVersionRequest$ = [3, n0, _SPVR,
    0,
    [_PVA, _MIU, _MIUa],
    [0, 1, 1], 2
];
exports.StartProjectVersionRequest$ = StartProjectVersionRequest$;
const StartProjectVersionResponse$ = [3, n0, _SPVRt,
    0,
    [_St],
    [0]
];
exports.StartProjectVersionResponse$ = StartProjectVersionResponse$;
const StartSegmentDetectionFilters$ = [3, n0, _SSDF,
    0,
    [_TCF, _SFh],
    [() => StartTechnicalCueDetectionFilter$, () => StartShotDetectionFilter$]
];
exports.StartSegmentDetectionFilters$ = StartSegmentDetectionFilters$;
const StartSegmentDetectionRequest$ = [3, n0, _SSDR,
    0,
    [_Vi, _STe, _CRT, _NC, _JT, _Fi],
    [() => Video$, 64 | 0, 0, () => NotificationChannel$, 0, () => StartSegmentDetectionFilters$], 2
];
exports.StartSegmentDetectionRequest$ = StartSegmentDetectionRequest$;
const StartSegmentDetectionResponse$ = [3, n0, _SSDRt,
    0,
    [_JI],
    [0]
];
exports.StartSegmentDetectionResponse$ = StartSegmentDetectionResponse$;
const StartShotDetectionFilter$ = [3, n0, _SSDFt,
    0,
    [_MSC],
    [1]
];
exports.StartShotDetectionFilter$ = StartShotDetectionFilter$;
const StartStreamProcessorRequest$ = [3, n0, _SSPR,
    0,
    [_N, _SSt, _SSto],
    [0, () => StreamProcessingStartSelector$, () => StreamProcessingStopSelector$], 1
];
exports.StartStreamProcessorRequest$ = StartStreamProcessorRequest$;
const StartStreamProcessorResponse$ = [3, n0, _SSPRt,
    0,
    [_SIe],
    [0]
];
exports.StartStreamProcessorResponse$ = StartStreamProcessorResponse$;
const StartTechnicalCueDetectionFilter$ = [3, n0, _STCDF,
    0,
    [_MSC, _BF],
    [1, () => BlackFrame$]
];
exports.StartTechnicalCueDetectionFilter$ = StartTechnicalCueDetectionFilter$;
const StartTextDetectionFilters$ = [3, n0, _STDF,
    0,
    [_WF, _ROI],
    [() => DetectionFilter$, () => RegionsOfInterest]
];
exports.StartTextDetectionFilters$ = StartTextDetectionFilters$;
const StartTextDetectionRequest$ = [3, n0, _STDR,
    0,
    [_Vi, _CRT, _NC, _JT, _Fi],
    [() => Video$, 0, () => NotificationChannel$, 0, () => StartTextDetectionFilters$], 1
];
exports.StartTextDetectionRequest$ = StartTextDetectionRequest$;
const StartTextDetectionResponse$ = [3, n0, _STDRt,
    0,
    [_JI],
    [0]
];
exports.StartTextDetectionResponse$ = StartTextDetectionResponse$;
const StopProjectVersionRequest$ = [3, n0, _SPVRto,
    0,
    [_PVA],
    [0], 1
];
exports.StopProjectVersionRequest$ = StopProjectVersionRequest$;
const StopProjectVersionResponse$ = [3, n0, _SPVRtop,
    0,
    [_St],
    [0]
];
exports.StopProjectVersionResponse$ = StopProjectVersionResponse$;
const StopStreamProcessorRequest$ = [3, n0, _SSPRto,
    0,
    [_N],
    [0], 1
];
exports.StopStreamProcessorRequest$ = StopStreamProcessorRequest$;
const StopStreamProcessorResponse$ = [3, n0, _SSPRtop,
    0,
    [],
    []
];
exports.StopStreamProcessorResponse$ = StopStreamProcessorResponse$;
const StreamProcessingStartSelector$ = [3, n0, _SPSS,
    0,
    [_KVSSSS],
    [() => KinesisVideoStreamStartSelector$]
];
exports.StreamProcessingStartSelector$ = StreamProcessingStartSelector$;
const StreamProcessingStopSelector$ = [3, n0, _SPSSt,
    0,
    [_MDIS],
    [1]
];
exports.StreamProcessingStopSelector$ = StreamProcessingStopSelector$;
const StreamProcessor$ = [3, n0, _SPt,
    0,
    [_N, _St],
    [0, 0]
];
exports.StreamProcessor$ = StreamProcessor$;
const StreamProcessorDataSharingPreference$ = [3, n0, _SPDSP,
    0,
    [_OI],
    [2], 1
];
exports.StreamProcessorDataSharingPreference$ = StreamProcessorDataSharingPreference$;
const StreamProcessorInput$ = [3, n0, _SPI,
    0,
    [_KVS],
    [() => KinesisVideoStream$]
];
exports.StreamProcessorInput$ = StreamProcessorInput$;
const StreamProcessorNotificationChannel$ = [3, n0, _SPNC,
    0,
    [_SNSTA],
    [0], 1
];
exports.StreamProcessorNotificationChannel$ = StreamProcessorNotificationChannel$;
const StreamProcessorOutput$ = [3, n0, _SPO,
    0,
    [_KDS, _SD],
    [() => KinesisDataStream$, () => S3Destination$]
];
exports.StreamProcessorOutput$ = StreamProcessorOutput$;
const StreamProcessorSettings$ = [3, n0, _SPS,
    0,
    [_FSa, _CH],
    [() => FaceSearchSettings$, () => ConnectedHomeSettings$]
];
exports.StreamProcessorSettings$ = StreamProcessorSettings$;
const StreamProcessorSettingsForUpdate$ = [3, n0, _SPSFU,
    0,
    [_CHFU],
    [() => ConnectedHomeSettingsForUpdate$]
];
exports.StreamProcessorSettingsForUpdate$ = StreamProcessorSettingsForUpdate$;
const Summary$ = [3, n0, _Su,
    0,
    [_SO],
    [() => S3Object$]
];
exports.Summary$ = Summary$;
const Sunglasses$ = [3, n0, _Sun,
    0,
    [_V, _Con],
    [2, 1]
];
exports.Sunglasses$ = Sunglasses$;
const TagResourceRequest$ = [3, n0, _TRR,
    0,
    [_RAe, _Ta],
    [0, 128 | 0], 2
];
exports.TagResourceRequest$ = TagResourceRequest$;
const TagResourceResponse$ = [3, n0, _TRRa,
    0,
    [],
    []
];
exports.TagResourceResponse$ = TagResourceResponse$;
const TechnicalCueSegment$ = [3, n0, _TCS,
    0,
    [_Ty, _Con],
    [0, 1]
];
exports.TechnicalCueSegment$ = TechnicalCueSegment$;
const TestingData$ = [3, n0, _TDe,
    0,
    [_As, _AC],
    [() => Assets, 2]
];
exports.TestingData$ = TestingData$;
const TestingDataResult$ = [3, n0, _TDRe,
    0,
    [_In, _O, _Va],
    [() => TestingData$, () => TestingData$, () => ValidationData$]
];
exports.TestingDataResult$ = TestingDataResult$;
const TextDetection$ = [3, n0, _TDext,
    0,
    [_DTe, _Ty, _I, _PIa, _Con, _G],
    [0, 0, 1, 1, 1, () => Geometry$]
];
exports.TextDetection$ = TextDetection$;
const TextDetectionResult$ = [3, n0, _TDRex,
    0,
    [_Ti, _TDext],
    [1, () => TextDetection$]
];
exports.TextDetectionResult$ = TextDetectionResult$;
const TrainingData$ = [3, n0, _TD,
    0,
    [_As],
    [() => Assets]
];
exports.TrainingData$ = TrainingData$;
const TrainingDataResult$ = [3, n0, _TDR,
    0,
    [_In, _O, _Va],
    [() => TrainingData$, () => TrainingData$, () => ValidationData$]
];
exports.TrainingDataResult$ = TrainingDataResult$;
const UnindexedFace$ = [3, n0, _UFni,
    0,
    [_Rea, _FDa],
    [64 | 0, () => FaceDetail$]
];
exports.UnindexedFace$ = UnindexedFace$;
const UnsearchedFace$ = [3, n0, _UFnse,
    0,
    [_FD, _Rea],
    [() => FaceDetail$, 64 | 0]
];
exports.UnsearchedFace$ = UnsearchedFace$;
const UnsuccessfulFaceAssociation$ = [3, n0, _UFAn,
    0,
    [_FI, _UI, _Con, _Rea],
    [0, 0, 1, 64 | 0]
];
exports.UnsuccessfulFaceAssociation$ = UnsuccessfulFaceAssociation$;
const UnsuccessfulFaceDeletion$ = [3, n0, _UFDns,
    0,
    [_FI, _UI, _Rea],
    [0, 0, 64 | 0]
];
exports.UnsuccessfulFaceDeletion$ = UnsuccessfulFaceDeletion$;
const UnsuccessfulFaceDisassociation$ = [3, n0, _UFDnsu,
    0,
    [_FI, _UI, _Rea],
    [0, 0, 64 | 0]
];
exports.UnsuccessfulFaceDisassociation$ = UnsuccessfulFaceDisassociation$;
const UntagResourceRequest$ = [3, n0, _URR,
    0,
    [_RAe, _TK],
    [0, 64 | 0], 2
];
exports.UntagResourceRequest$ = UntagResourceRequest$;
const UntagResourceResponse$ = [3, n0, _URRn,
    0,
    [],
    []
];
exports.UntagResourceResponse$ = UntagResourceResponse$;
const UpdateDatasetEntriesRequest$ = [3, n0, _UDER,
    0,
    [_DA, _Cha],
    [0, () => DatasetChanges$], 2
];
exports.UpdateDatasetEntriesRequest$ = UpdateDatasetEntriesRequest$;
const UpdateDatasetEntriesResponse$ = [3, n0, _UDERp,
    0,
    [],
    []
];
exports.UpdateDatasetEntriesResponse$ = UpdateDatasetEntriesResponse$;
const UpdateStreamProcessorRequest$ = [3, n0, _USPR,
    0,
    [_N, _SFU, _ROIFU, _DSPFU, _PTD],
    [0, () => StreamProcessorSettingsForUpdate$, () => RegionsOfInterest, () => StreamProcessorDataSharingPreference$, 64 | 0], 1
];
exports.UpdateStreamProcessorRequest$ = UpdateStreamProcessorRequest$;
const UpdateStreamProcessorResponse$ = [3, n0, _USPRp,
    0,
    [],
    []
];
exports.UpdateStreamProcessorResponse$ = UpdateStreamProcessorResponse$;
const User$ = [3, n0, _Use,
    0,
    [_UI, _US],
    [0, 0]
];
exports.User$ = User$;
const UserMatch$ = [3, n0, _UMs,
    0,
    [_Si, _Use],
    [1, () => MatchedUser$]
];
exports.UserMatch$ = UserMatch$;
const ValidationData$ = [3, n0, _VDa,
    0,
    [_As],
    [() => Assets]
];
exports.ValidationData$ = ValidationData$;
const Versions$ = [3, n0, _Ver,
    0,
    [_Mi, _Ma],
    [0, 0]
];
exports.Versions$ = Versions$;
const Video$ = [3, n0, _Vi,
    0,
    [_SO],
    [() => S3Object$]
];
exports.Video$ = Video$;
const VideoMetadata$ = [3, n0, _VM,
    0,
    [_Co, _DM, _For, _FRr, _FH, _FW, _CRo],
    [0, 1, 0, 1, 1, 1, 0]
];
exports.VideoMetadata$ = VideoMetadata$;
var Assets = [1, n0, _As,
    0, () => Asset$
];
var AssociatedFacesList = [1, n0, _AFL,
    0, () => AssociatedFace$
];
var Attributes = 64 | 0;
var AudioMetadataList = [1, n0, _AML,
    0, () => AudioMetadata$
];
var AuditImages = [1, n0, _AIu,
    0, [() => AuditImage$,
        0]
];
var BodyParts = [1, n0, _BP,
    0, () => ProtectiveEquipmentBodyPart$
];
var CelebrityList = [1, n0, _CLe,
    0, () => Celebrity$
];
var CelebrityRecognitions = [1, n0, _CRe,
    0, () => CelebrityRecognition$
];
var ChallengePreferences = [1, n0, _CPh,
    0, () => ChallengePreference$
];
var CollectionIdList = 64 | 0;
var ComparedFaceList = [1, n0, _CFL,
    0, () => ComparedFace$
];
var CompareFacesMatchList = [1, n0, _CFML,
    0, () => CompareFacesMatch$
];
var CompareFacesUnmatchList = [1, n0, _CFUL,
    0, () => ComparedFace$
];
var ConnectedHomeLabels = 64 | 0;
var ContentClassifiers = 64 | 0;
var ContentModerationDetections = [1, n0, _CMDo,
    0, () => ContentModerationDetection$
];
var ContentTypes = [1, n0, _CT,
    0, () => ContentType$
];
var CustomizationFeatures = 64 | 0;
var CustomLabels = [1, n0, _CLu,
    0, () => CustomLabel$
];
var DatasetEntries = 64 | 0;
var DatasetLabelDescriptions = [1, n0, _DLDa,
    0, () => DatasetLabelDescription$
];
var DatasetLabels = 64 | 0;
var DatasetMetadataList = [1, n0, _DMLa,
    0, () => DatasetMetadata$
];
var DetectLabelsFeatureList = 64 | 0;
var DisassociatedFacesList = [1, n0, _DFL,
    0, () => DisassociatedFace$
];
var DistributeDatasetMetadataList = [1, n0, _DDML,
    0, () => DistributeDataset$
];
var DominantColors = [1, n0, _DCo,
    0, () => DominantColor$
];
var Emotions = [1, n0, _E,
    0, () => Emotion$
];
var EquipmentDetections = [1, n0, _EDq,
    0, () => EquipmentDetection$
];
var FaceDetailList = [1, n0, _FDL,
    0, () => FaceDetail$
];
var FaceDetections = [1, n0, _FDace,
    0, () => FaceDetection$
];
var FaceIdList = 64 | 0;
var FaceList = [1, n0, _FL,
    0, () => Face$
];
var FaceMatchList = [1, n0, _FML,
    0, () => FaceMatch$
];
var FaceModelVersionList = 64 | 0;
var FaceRecordList = [1, n0, _FRL,
    0, () => FaceRecord$
];
var GeneralLabelsFilterList = 64 | 0;
var HumanLoopActivationReasons = 64 | 0;
var Instances = [1, n0, _Inst,
    0, () => Instance$
];
var LabelAliases = [1, n0, _LAa,
    0, () => LabelAlias$
];
var LabelCategories = [1, n0, _LCa,
    0, () => LabelCategory$
];
var LabelDetectionFeatureList = 64 | 0;
var LabelDetections = [1, n0, _LDa,
    0, () => LabelDetection$
];
var Labels = [1, n0, _Lab,
    0, () => Label$
];
var Landmarks = [1, n0, _La,
    0, () => Landmark$
];
var MediaAnalysisJobDescriptions = [1, n0, _MAJDe,
    0, () => MediaAnalysisJobDescription$
];
var ModerationLabels = [1, n0, _MLo,
    0, () => ModerationLabel$
];
var Parents = [1, n0, _Pa,
    0, () => Parent$
];
var PersonDetections = [1, n0, _PDers,
    0, () => PersonDetection$
];
var PersonMatches = [1, n0, _PMe,
    0, () => PersonMatch$
];
var Polygon = [1, n0, _Po,
    0, () => Point$
];
var ProjectDescriptions = [1, n0, _PD,
    0, () => ProjectDescription$
];
var ProjectNames = 64 | 0;
var ProjectPolicies = [1, n0, _PPr,
    0, () => ProjectPolicy$
];
var ProjectVersionDescriptions = [1, n0, _PVD,
    0, () => ProjectVersionDescription$
];
var ProtectiveEquipmentPersonIds = 64 | 1;
var ProtectiveEquipmentPersons = [1, n0, _PEPr,
    0, () => ProtectiveEquipmentPerson$
];
var ProtectiveEquipmentTypes = 64 | 0;
var Reasons = 64 | 0;
var RegionsOfInterest = [1, n0, _ROI,
    0, () => RegionOfInterest$
];
var SegmentDetections = [1, n0, _SDeg,
    0, () => SegmentDetection$
];
var SegmentTypes = 64 | 0;
var SegmentTypesInfo = [1, n0, _STIe,
    0, () => SegmentTypeInfo$
];
var StreamProcessorList = [1, n0, _SPL,
    0, () => StreamProcessor$
];
var StreamProcessorParametersToDelete = 64 | 0;
var TagKeyList = 64 | 0;
var TextDetectionList = [1, n0, _TDL,
    0, () => TextDetection$
];
var TextDetectionResults = [1, n0, _TDRext,
    0, () => TextDetectionResult$
];
var UnindexedFaces = [1, n0, _UFn,
    0, () => UnindexedFace$
];
var UnsearchedFaceReasons = 64 | 0;
var UnsearchedFacesList = [1, n0, _UFL,
    0, () => UnsearchedFace$
];
var UnsuccessfulFaceAssociationList = [1, n0, _UFAL,
    0, () => UnsuccessfulFaceAssociation$
];
var UnsuccessfulFaceAssociationReasons = 64 | 0;
var UnsuccessfulFaceDeletionReasons = 64 | 0;
var UnsuccessfulFaceDeletionsList = [1, n0, _UFDL,
    0, () => UnsuccessfulFaceDeletion$
];
var UnsuccessfulFaceDisassociationList = [1, n0, _UFDLn,
    0, () => UnsuccessfulFaceDisassociation$
];
var UnsuccessfulFaceDisassociationReasons = 64 | 0;
var Urls = 64 | 0;
var UserFaceIdList = 64 | 0;
var UserList = [1, n0, _UL,
    0, () => User$
];
var UserMatchList = [1, n0, _UML,
    0, () => UserMatch$
];
var VersionNames = 64 | 0;
var VideoMetadataList = [1, n0, _VML,
    0, () => VideoMetadata$
];
var TagMap = 128 | 0;
exports.AssociateFaces$ = [9, n0, _AFss,
    0, () => AssociateFacesRequest$, () => AssociateFacesResponse$
];
exports.CompareFaces$ = [9, n0, _CFo,
    0, () => CompareFacesRequest$, () => CompareFacesResponse$
];
exports.CopyProjectVersion$ = [9, n0, _CPV,
    0, () => CopyProjectVersionRequest$, () => CopyProjectVersionResponse$
];
exports.CreateCollection$ = [9, n0, _CCr,
    0, () => CreateCollectionRequest$, () => CreateCollectionResponse$
];
exports.CreateDataset$ = [9, n0, _CDr,
    0, () => CreateDatasetRequest$, () => CreateDatasetResponse$
];
exports.CreateFaceLivenessSession$ = [9, n0, _CFLS,
    2, () => CreateFaceLivenessSessionRequest$, () => CreateFaceLivenessSessionResponse$
];
exports.CreateProject$ = [9, n0, _CPr,
    0, () => CreateProjectRequest$, () => CreateProjectResponse$
];
exports.CreateProjectVersion$ = [9, n0, _CPVr,
    0, () => CreateProjectVersionRequest$, () => CreateProjectVersionResponse$
];
exports.CreateStreamProcessor$ = [9, n0, _CSP,
    0, () => CreateStreamProcessorRequest$, () => CreateStreamProcessorResponse$
];
exports.CreateUser$ = [9, n0, _CU,
    0, () => CreateUserRequest$, () => CreateUserResponse$
];
exports.DeleteCollection$ = [9, n0, _DCe,
    0, () => DeleteCollectionRequest$, () => DeleteCollectionResponse$
];
exports.DeleteDataset$ = [9, n0, _DDe,
    0, () => DeleteDatasetRequest$, () => DeleteDatasetResponse$
];
exports.DeleteFaces$ = [9, n0, _DFel,
    0, () => DeleteFacesRequest$, () => DeleteFacesResponse$
];
exports.DeleteProject$ = [9, n0, _DP,
    0, () => DeleteProjectRequest$, () => DeleteProjectResponse$
];
exports.DeleteProjectPolicy$ = [9, n0, _DPP,
    0, () => DeleteProjectPolicyRequest$, () => DeleteProjectPolicyResponse$
];
exports.DeleteProjectVersion$ = [9, n0, _DPV,
    0, () => DeleteProjectVersionRequest$, () => DeleteProjectVersionResponse$
];
exports.DeleteStreamProcessor$ = [9, n0, _DSPe,
    0, () => DeleteStreamProcessorRequest$, () => DeleteStreamProcessorResponse$
];
exports.DeleteUser$ = [9, n0, _DU,
    0, () => DeleteUserRequest$, () => DeleteUserResponse$
];
exports.DescribeCollection$ = [9, n0, _DCes,
    0, () => DescribeCollectionRequest$, () => DescribeCollectionResponse$
];
exports.DescribeDataset$ = [9, n0, _DDes,
    0, () => DescribeDatasetRequest$, () => DescribeDatasetResponse$
];
exports.DescribeProjects$ = [9, n0, _DPe,
    0, () => DescribeProjectsRequest$, () => DescribeProjectsResponse$
];
exports.DescribeProjectVersions$ = [9, n0, _DPVe,
    0, () => DescribeProjectVersionsRequest$, () => DescribeProjectVersionsResponse$
];
exports.DescribeStreamProcessor$ = [9, n0, _DSPes,
    0, () => DescribeStreamProcessorRequest$, () => DescribeStreamProcessorResponse$
];
exports.DetectCustomLabels$ = [9, n0, _DCL,
    0, () => DetectCustomLabelsRequest$, () => DetectCustomLabelsResponse$
];
exports.DetectFaces$ = [9, n0, _DFet,
    0, () => DetectFacesRequest$, () => DetectFacesResponse$
];
exports.DetectLabels$ = [9, n0, _DL,
    0, () => DetectLabelsRequest$, () => DetectLabelsResponse$
];
exports.DetectModerationLabels$ = [9, n0, _DML,
    0, () => DetectModerationLabelsRequest$, () => DetectModerationLabelsResponse$
];
exports.DetectProtectiveEquipment$ = [9, n0, _DPE,
    0, () => DetectProtectiveEquipmentRequest$, () => DetectProtectiveEquipmentResponse$
];
exports.DetectText$ = [9, n0, _DTet,
    0, () => DetectTextRequest$, () => DetectTextResponse$
];
exports.DisassociateFaces$ = [9, n0, _DFisa,
    0, () => DisassociateFacesRequest$, () => DisassociateFacesResponse$
];
exports.DistributeDatasetEntries$ = [9, n0, _DDE,
    0, () => DistributeDatasetEntriesRequest$, () => DistributeDatasetEntriesResponse$
];
exports.GetCelebrityInfo$ = [9, n0, _GCI,
    0, () => GetCelebrityInfoRequest$, () => GetCelebrityInfoResponse$
];
exports.GetCelebrityRecognition$ = [9, n0, _GCR,
    0, () => GetCelebrityRecognitionRequest$, () => GetCelebrityRecognitionResponse$
];
exports.GetContentModeration$ = [9, n0, _GCM,
    0, () => GetContentModerationRequest$, () => GetContentModerationResponse$
];
exports.GetFaceDetection$ = [9, n0, _GFD,
    0, () => GetFaceDetectionRequest$, () => GetFaceDetectionResponse$
];
exports.GetFaceLivenessSessionResults$ = [9, n0, _GFLSR,
    0, () => GetFaceLivenessSessionResultsRequest$, () => GetFaceLivenessSessionResultsResponse$
];
exports.GetFaceSearch$ = [9, n0, _GFS,
    0, () => GetFaceSearchRequest$, () => GetFaceSearchResponse$
];
exports.GetLabelDetection$ = [9, n0, _GLD,
    0, () => GetLabelDetectionRequest$, () => GetLabelDetectionResponse$
];
exports.GetMediaAnalysisJob$ = [9, n0, _GMAJ,
    0, () => GetMediaAnalysisJobRequest$, () => GetMediaAnalysisJobResponse$
];
exports.GetPersonTracking$ = [9, n0, _GPT,
    0, () => GetPersonTrackingRequest$, () => GetPersonTrackingResponse$
];
exports.GetSegmentDetection$ = [9, n0, _GSD,
    0, () => GetSegmentDetectionRequest$, () => GetSegmentDetectionResponse$
];
exports.GetTextDetection$ = [9, n0, _GTD,
    0, () => GetTextDetectionRequest$, () => GetTextDetectionResponse$
];
exports.IndexFaces$ = [9, n0, _IF,
    0, () => IndexFacesRequest$, () => IndexFacesResponse$
];
exports.ListCollections$ = [9, n0, _LCi,
    0, () => ListCollectionsRequest$, () => ListCollectionsResponse$
];
exports.ListDatasetEntries$ = [9, n0, _LDE,
    0, () => ListDatasetEntriesRequest$, () => ListDatasetEntriesResponse$
];
exports.ListDatasetLabels$ = [9, n0, _LDL,
    0, () => ListDatasetLabelsRequest$, () => ListDatasetLabelsResponse$
];
exports.ListFaces$ = [9, n0, _LF,
    0, () => ListFacesRequest$, () => ListFacesResponse$
];
exports.ListMediaAnalysisJobs$ = [9, n0, _LMAJ,
    0, () => ListMediaAnalysisJobsRequest$, () => ListMediaAnalysisJobsResponse$
];
exports.ListProjectPolicies$ = [9, n0, _LPP,
    0, () => ListProjectPoliciesRequest$, () => ListProjectPoliciesResponse$
];
exports.ListStreamProcessors$ = [9, n0, _LSP,
    0, () => ListStreamProcessorsRequest$, () => ListStreamProcessorsResponse$
];
exports.ListTagsForResource$ = [9, n0, _LTFR,
    0, () => ListTagsForResourceRequest$, () => ListTagsForResourceResponse$
];
exports.ListUsers$ = [9, n0, _LU,
    0, () => ListUsersRequest$, () => ListUsersResponse$
];
exports.PutProjectPolicy$ = [9, n0, _PPP,
    0, () => PutProjectPolicyRequest$, () => PutProjectPolicyResponse$
];
exports.RecognizeCelebrities$ = [9, n0, _RC,
    0, () => RecognizeCelebritiesRequest$, () => RecognizeCelebritiesResponse$
];
exports.SearchFaces$ = [9, n0, _SFe,
    0, () => SearchFacesRequest$, () => SearchFacesResponse$
];
exports.SearchFacesByImage$ = [9, n0, _SFBI,
    0, () => SearchFacesByImageRequest$, () => SearchFacesByImageResponse$
];
exports.SearchUsers$ = [9, n0, _SUe,
    0, () => SearchUsersRequest$, () => SearchUsersResponse$
];
exports.SearchUsersByImage$ = [9, n0, _SUBI,
    0, () => SearchUsersByImageRequest$, () => SearchUsersByImageResponse$
];
exports.StartCelebrityRecognition$ = [9, n0, _SCR,
    2, () => StartCelebrityRecognitionRequest$, () => StartCelebrityRecognitionResponse$
];
exports.StartContentModeration$ = [9, n0, _SCM,
    2, () => StartContentModerationRequest$, () => StartContentModerationResponse$
];
exports.StartFaceDetection$ = [9, n0, _SFDt,
    2, () => StartFaceDetectionRequest$, () => StartFaceDetectionResponse$
];
exports.StartFaceSearch$ = [9, n0, _SFS,
    2, () => StartFaceSearchRequest$, () => StartFaceSearchResponse$
];
exports.StartLabelDetection$ = [9, n0, _SLD,
    2, () => StartLabelDetectionRequest$, () => StartLabelDetectionResponse$
];
exports.StartMediaAnalysisJob$ = [9, n0, _SMAJ,
    2, () => StartMediaAnalysisJobRequest$, () => StartMediaAnalysisJobResponse$
];
exports.StartPersonTracking$ = [9, n0, _SPT,
    2, () => StartPersonTrackingRequest$, () => StartPersonTrackingResponse$
];
exports.StartProjectVersion$ = [9, n0, _SPV,
    0, () => StartProjectVersionRequest$, () => StartProjectVersionResponse$
];
exports.StartSegmentDetection$ = [9, n0, _SSD,
    2, () => StartSegmentDetectionRequest$, () => StartSegmentDetectionResponse$
];
exports.StartStreamProcessor$ = [9, n0, _SSP,
    0, () => StartStreamProcessorRequest$, () => StartStreamProcessorResponse$
];
exports.StartTextDetection$ = [9, n0, _STD,
    2, () => StartTextDetectionRequest$, () => StartTextDetectionResponse$
];
exports.StopProjectVersion$ = [9, n0, _SPVt,
    0, () => StopProjectVersionRequest$, () => StopProjectVersionResponse$
];
exports.StopStreamProcessor$ = [9, n0, _SSPt,
    0, () => StopStreamProcessorRequest$, () => StopStreamProcessorResponse$
];
exports.TagResource$ = [9, n0, _TR,
    0, () => TagResourceRequest$, () => TagResourceResponse$
];
exports.UntagResource$ = [9, n0, _UR,
    0, () => UntagResourceRequest$, () => UntagResourceResponse$
];
exports.UpdateDatasetEntries$ = [9, n0, _UDE,
    0, () => UpdateDatasetEntriesRequest$, () => UpdateDatasetEntriesResponse$
];
exports.UpdateStreamProcessor$ = [9, n0, _USP,
    0, () => UpdateStreamProcessorRequest$, () => UpdateStreamProcessorResponse$
];
